{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "import numpy as np\n",
    "import os\n",
    "import random"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ideas of the program would be:\n",
    "\n",
    "- want to read in reference_dataframe.csv\n",
    "\n",
    "- taxonomic level e.g. genus=Candida -> species level comparision\n",
    "\n",
    "- path to location of data. Root of analysis/....."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With arg parse\n",
    "\n",
    "ref_df_fn = args.ref_df_fn  \n",
    "data_root = args.data_root  \n",
    "tax_rank = args.tax_rank #use with flag e.g. -r  \n",
    "name = args.name #name of the rank to select from, use flag e.g. -n  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df_fn = '../../analysis/Stats/reference_dataframe.csv'\n",
    "ref_df_fn = os.path.abspath(ref_df_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "try:\n",
    "    os.path.exists(ref_df_fn)\n",
    "except:\n",
    "    print('Cannot find %s' % ref_df_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#to be able to handle anything from command lien\n",
    "tax_rank = 'family'\n",
    "name = 'Saccharomycetaceae'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_reads = 15000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "name = name.lower()\n",
    "tax_rank = tax_rank.lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df = pd.read_csv(ref_df_fn, index_col=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df[ref_df[\"# reads after length filtering\"] < n_reads].shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "try:\n",
    "    if ref_df[ref_df[\"# reads after length filtering\"] \\\n",
    "              < n_reads].shape[0] >0 :\n",
    "        print(\"These species need more reads.\")\n",
    "        print(ref_df[ref_df[\"# reads after length filtering\"] \\\n",
    "              < n_reads])\n",
    "        #exit()\n",
    "except:\n",
    "    print('Check %s to have the wanted column names' % ref_df_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#not neccessary pull from command line\n",
    "# name = \"Trichocomaceae\"\n",
    "data_root = '../../'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "indecies = ref_df[ref_df[tax_rank] == name].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "indecies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "SeqIO_dicts = {}\n",
    "for index in indecies:\n",
    "    fasta_path = os.path.join(data_root, ref_df.loc[index, 'path to length filtering'])\n",
    "    print(fasta_path)\n",
    "    try:\n",
    "        SeqIO_dicts[index] = SeqIO.to_dict(SeqIO.parse(fasta_path, 'fasta'))\n",
    "    except:\n",
    "        print('Check location of fasta files.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def max_seq_len(SeqIO_dict):\n",
    "    \"\"\"\n",
    "    Fucntion takes a SeqIO_dict and returns the lenghts of the\n",
    "    longest sequence.\n",
    "    \"\"\"\n",
    "    total_lens = []\n",
    "    for key in SeqIO_dict.keys():\n",
    "        total_lens.append(len(SeqIO_dict[key].seq))\n",
    "    return max(total_lens)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def numberfi(SeqIO_dict, seq_len, nsubsample):\n",
    "    \"\"\"\n",
    "    Take SeqIO_dict and return SeqIO_dict were bases have been replaced\n",
    "    with numbers\n",
    "    ACGT- to 01234\n",
    "    Take the seq_len each sequence should have.\n",
    "    \"\"\"\n",
    "    num_dict = {}\n",
    "    \n",
    "    keys = list(SeqIO_dict.keys())\n",
    "    randkeys = random.sample(keys, k= nsubsample)\n",
    "    \n",
    "    \n",
    "    for key in randkeys:\n",
    "        seq = str(SeqIO_dict[key].seq).replace(\"A\",'0 ')\\\n",
    "        .replace(\"C\",'1 ').replace(\"G\",'2 ').replace(\"T\",'3 ')\n",
    "        seq_new = seq + '4 '*(seq_len -int(len(seq)/2))\n",
    "        num_dict[key] = list(map(int, seq_new.split(' ')[:-1]))\n",
    "    return num_dict\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_lens = []\n",
    "for key, value in SeqIO_dicts.items():\n",
    "    total_lens.append(max_seq_len(value))\n",
    "print(\"This is the maximum sequence length of all sampled sequences.\")\n",
    "print(total_lens)\n",
    "\n",
    "numSeqIO_dicts = {}\n",
    "max_len = max(total_lens)\n",
    "for key, value in SeqIO_dicts.items():\n",
    "    numSeqIO_dicts[key] = numberfi(value, max_len, s_reads)\n",
    "\n",
    "seq_list = []\n",
    "for index in indecies:\n",
    "    seq_list.append(np.array(list(numSeqIO_dicts[index].values())))\n",
    "seq_comb = np.concatenate(seq_list, axis = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_class = len(numSeqIO_dicts.keys())\n",
    "ids_comb = np.zeros( (n_reads*num_class,num_class) )\n",
    "for i in range(0, num_class):\n",
    "    ids_comb[i*n_reads:(i+1)*n_reads,i] = 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_class = 5\n",
    "ids_comb = np.zeros( (n_reads*num_class,num_class) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(0, num_class):\n",
    "    ids_comb[i*n_reads:(i+1)*n_reads,i] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ids_comb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df[tax_rank] == 'Candida'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "location = (ref_df.columns.get_loc(tax_rank)-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "classes = ref_df.iloc[indecies,location].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_dict = {}\n",
    "for item in classes:\n",
    "    count_dict[item] = sum(ref_df.iloc[:,location] == item)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "classes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for item in count_dict.keys():\n",
    "    print(n_reads/count_dict[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# define n_reads\n",
    "\n",
    "min_vals = []\n",
    "for key, value in count_dict.items():\n",
    "    if value == min(count_dict.values()):\n",
    "        min_vals.append(ref_df[ref_df.iloc[:,location] == item]['# reads after length filtering'].min())\n",
    "min(min_vals)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "max_reads = 0\n",
    "for key, value in count_dict.items():\n",
    "    if value == max(count_dict.values()):\n",
    "        max_reads = value*n_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "max_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "min_vals = [44999]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# is max_reads < min_reads?\n",
    "# if yes, use max_reads to subsample as base value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if max_reads <= min(min_vals):\n",
    "    for key, value in count_dict.items():\n",
    "        s_reads = int(max_reads/value)\n",
    "        print(key, s_reads)\n",
    "else:\n",
    "    for key, value in count_dict.items():\n",
    "        s_reads = int(min(min_vals)/value)\n",
    "        print(key, s_reads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "location = (ref_df.columns.get_loc(tax_rank)-1)\n",
    "col_name = ref_df.columns[location]\n",
    "\n",
    "classes = ref_df.iloc[indecies,location].unique()\n",
    "\n",
    "count_dict = {}\n",
    "for item in classes:\n",
    "    count_dict[item] = sum(ref_df.iloc[:,location] == item)\n",
    "\n",
    "min_vals = []\n",
    "for key, value in count_dict.items():\n",
    "    if value == min(count_dict.values()):\n",
    "        min_vals.append(ref_df[ref_df.iloc[:,location] == item]['# reads after length filtering'].min())\n",
    "minimum_value = min(min_vals)\n",
    "\n",
    "max_reads = 0\n",
    "for key, value in count_dict.items():\n",
    "    if value == max(count_dict.values()):\n",
    "        max_reads = value*n_reads\n",
    "\n",
    "if max_reads <= minimum_value:\n",
    "    minimum_value = max_reads\n",
    "for key, n_class in count_dict.items():\n",
    "    s_reads = int(minimum_value/n_class)\n",
    "    if ref_df[ref_df.loc[:,col_name]==key]['# reads after length filtering'].min() < s_reads:\n",
    "        minimum_value = ref_df[ref_df.loc[:,col_name]==key]['# reads after length filtering'].min()/n_class\n",
    "        s_reads = int(minimum_value/n_class)\n",
    "    print(key, s_reads)\n",
    "    \n",
    "# parse s_reads into numberfy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['genus', 'family', 'order', 'class', 'phylum', 'kingdom'], dtype='object')\n"
     ]
    }
   ],
   "source": [
    "tax_ranks = ref_df.columns[2:-6]\n",
    "print(tax_ranks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "genus\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "puccinia\n",
      "['striiformis-tritici']\n",
      "{'striiformis-tritici': 1}\n",
      "minimum_value = 113336\n",
      "15000\n",
      "\n",
      "\n",
      "zymoseptoria\n",
      "['tritici']\n",
      "{'tritici': 1}\n",
      "minimum_value = 133088\n",
      "15000\n",
      "\n",
      "\n",
      "pyrenophora\n",
      "['tritici-repentis']\n",
      "{'tritici-repentis': 1}\n",
      "minimum_value = 90014\n",
      "15000\n",
      "\n",
      "\n",
      "fusarium\n",
      "['oxysporum']\n",
      "{'oxysporum': 1}\n",
      "minimum_value = 123742\n",
      "15000\n",
      "\n",
      "\n",
      "tuber\n",
      "['brumale']\n",
      "{'brumale': 1}\n",
      "minimum_value = 74232\n",
      "15000\n",
      "\n",
      "\n",
      "cortinarius\n",
      "['globuliformis']\n",
      "{'globuliformis': 1}\n",
      "minimum_value = 117090\n",
      "15000\n",
      "\n",
      "\n",
      "aspergillus\n",
      "['niger' 'flavus' 'unidentified']\n",
      "{'niger': 1, 'flavus': 1, 'unidentified': 1}\n",
      "minimum_value = 25916\n",
      "niger 15000\n",
      "flavus 15000\n",
      "unidentified 15000\n",
      "[15000, 15000, 15000]\n",
      "\n",
      "\n",
      "clavispora\n",
      "['lusitaniae']\n",
      "{'lusitaniae': 1}\n",
      "minimum_value = 131936\n",
      "15000\n",
      "\n",
      "\n",
      "kluyveromyces\n",
      "['unidentified' 'marxianus']\n",
      "{'unidentified': 1, 'marxianus': 1}\n",
      "minimum_value = 25916\n",
      "unidentified 15000\n",
      "marxianus 15000\n",
      "[15000, 15000]\n",
      "\n",
      "\n",
      "penicillium\n",
      "['chrysogenum']\n",
      "{'chrysogenum': 1}\n",
      "minimum_value = 72306\n",
      "15000\n",
      "\n",
      "\n",
      "rhodotorula\n",
      "['mucilaginosa']\n",
      "{'mucilaginosa': 1}\n",
      "minimum_value = 117800\n",
      "15000\n",
      "\n",
      "\n",
      "scedosporium\n",
      "['boydii']\n",
      "{'boydii': 1}\n",
      "minimum_value = 93722\n",
      "15000\n",
      "\n",
      "\n",
      "blastobotrys\n",
      "['proliferans']\n",
      "{'proliferans': 1}\n",
      "minimum_value = 37314\n",
      "15000\n",
      "\n",
      "\n",
      "debaryomyces\n",
      "['unidentified']\n",
      "{'unidentified': 1}\n",
      "minimum_value = 25916\n",
      "15000\n",
      "\n",
      "\n",
      "galactomyces\n",
      "['geotrichum']\n",
      "{'geotrichum': 1}\n",
      "minimum_value = 7804\n",
      "15000\n",
      "\n",
      "\n",
      "kodamaea\n",
      "['ohmeri']\n",
      "{'ohmeri': 1}\n",
      "minimum_value = 36478\n",
      "15000\n",
      "\n",
      "\n",
      "meyerozyma\n",
      "['guillermondii']\n",
      "{'guillermondii': 1}\n",
      "minimum_value = 18944\n",
      "15000\n",
      "\n",
      "\n",
      "wickerhamomyces\n",
      "['anomalus']\n",
      "{'anomalus': 1}\n",
      "minimum_value = 42588\n",
      "15000\n",
      "\n",
      "\n",
      "yamadazyma\n",
      "['mexicana' 'scolyti']\n",
      "{'mexicana': 1, 'scolyti': 1}\n",
      "minimum_value = 34840\n",
      "mexicana 15000\n",
      "scolyti 15000\n",
      "[15000, 15000]\n",
      "\n",
      "\n",
      "yarrowia\n",
      "['lipolytica']\n",
      "{'lipolytica': 1}\n",
      "minimum_value = 33872\n",
      "15000\n",
      "\n",
      "\n",
      "zygoascus\n",
      "['hellenicus']\n",
      "{'hellenicus': 1}\n",
      "minimum_value = 34002\n",
      "15000\n",
      "\n",
      "\n",
      "cryptococcus\n",
      "['zero' 'gattii']\n",
      "{'zero': 1, 'gattii': 1}\n",
      "minimum_value = 23660\n",
      "zero 15000\n",
      "gattii 15000\n",
      "[15000, 15000]\n",
      "\n",
      "\n",
      "diaporthe\n",
      "['ccl067' 'unidentified']\n",
      "{'ccl067': 1, 'unidentified': 1}\n",
      "minimum_value = 25916\n",
      "ccl067 15000\n",
      "unidentified 15000\n",
      "[15000, 15000]\n",
      "\n",
      "\n",
      "oculimacula\n",
      "['yallundae-ccl031' 'yallundae-ccl029']\n",
      "{'yallundae-ccl031': 1, 'yallundae-ccl029': 1}\n",
      "minimum_value = 49480\n",
      "yallundae-ccl031 15000\n",
      "yallundae-ccl029 15000\n",
      "[15000, 15000]\n",
      "\n",
      "\n",
      "dothiorella\n",
      "['vidmadera']\n",
      "{'vidmadera': 1}\n",
      "minimum_value = 44256\n",
      "15000\n",
      "\n",
      "\n",
      "quambalaria\n",
      "['cyanescens']\n",
      "{'cyanescens': 1}\n",
      "minimum_value = 46170\n",
      "15000\n",
      "\n",
      "\n",
      "entoleuca\n",
      "['unidentified']\n",
      "{'unidentified': 1}\n",
      "minimum_value = 25916\n",
      "15000\n",
      "\n",
      "\n",
      "asteroma\n",
      "['ccl060' 'ccl068']\n",
      "{'ccl060': 1, 'ccl068': 1}\n",
      "minimum_value = 31996\n",
      "ccl060 15000\n",
      "ccl068 15000\n",
      "[15000, 15000]\n",
      "\n",
      "\n",
      "saccharomyces\n",
      "['cerevisiae']\n",
      "{'cerevisiae': 1}\n",
      "minimum_value = 30260\n",
      "15000\n",
      "\n",
      "\n",
      "cladophialophora\n",
      "['unidentified']\n",
      "{'unidentified': 1}\n",
      "minimum_value = 25916\n",
      "15000\n",
      "\n",
      "\n",
      "candida\n",
      "['albicans' 'metapsilosis' 'orthopsilosis' 'parapsilosis' 'unidentified']\n",
      "{'albicans': 1, 'metapsilosis': 1, 'orthopsilosis': 1, 'parapsilosis': 1, 'unidentified': 1}\n",
      "minimum_value = 25596\n",
      "albicans 15000\n",
      "metapsilosis 15000\n",
      "orthopsilosis 15000\n",
      "parapsilosis 15000\n",
      "unidentified 15000\n",
      "[15000, 15000, 15000, 15000, 15000]\n",
      "\n",
      "\n",
      "geotrichum\n",
      "['candidum']\n",
      "{'candidum': 1}\n",
      "minimum_value = 27094\n",
      "15000\n",
      "\n",
      "\n",
      "pichia\n",
      "['kudriavzevii' 'membranifaciens']\n",
      "{'kudriavzevii': 1, 'membranifaciens': 1}\n",
      "minimum_value = 26936\n",
      "kudriavzevii 15000\n",
      "membranifaciens 15000\n",
      "[15000, 15000]\n",
      "family\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "pucciniaceae\n",
      "['puccinia']\n",
      "{'puccinia': 1}\n",
      "minimum_value = 113336\n",
      "15000\n",
      "\n",
      "\n",
      "mycosphaerellaceae\n",
      "['zymoseptoria']\n",
      "{'zymoseptoria': 1}\n",
      "minimum_value = 133088\n",
      "15000\n",
      "\n",
      "\n",
      "pleosporaceae\n",
      "['pyrenophora']\n",
      "{'pyrenophora': 1}\n",
      "minimum_value = 90014\n",
      "15000\n",
      "\n",
      "\n",
      "nectriaceae\n",
      "['fusarium']\n",
      "{'fusarium': 1}\n",
      "minimum_value = 123742\n",
      "15000\n",
      "\n",
      "\n",
      "tuberaceae\n",
      "['tuber']\n",
      "{'tuber': 1}\n",
      "minimum_value = 74232\n",
      "15000\n",
      "\n",
      "\n",
      "cortinariaceae\n",
      "['cortinarius']\n",
      "{'cortinarius': 1}\n",
      "minimum_value = 117090\n",
      "15000\n",
      "\n",
      "\n",
      "trichocomaceae\n",
      "['aspergillus' 'penicillium']\n",
      "{'aspergillus': 3, 'penicillium': 1}\n",
      "minimum_value = 72306\n",
      "aspergillus 15000\n",
      "penicillium 45000\n",
      "[45000, 45000]\n",
      "\n",
      "\n",
      "metschnikowiaceae\n",
      "['clavispora' 'kodamaea']\n",
      "{'clavispora': 1, 'kodamaea': 1}\n",
      "minimum_value = 36478\n",
      "clavispora 15000\n",
      "kodamaea 15000\n",
      "[15000, 15000]\n",
      "\n",
      "\n",
      "saccharomycetaceae\n",
      "['kluyveromyces' 'saccharomyces' 'candida']\n",
      "{'kluyveromyces': 2, 'saccharomyces': 1, 'candida': 5}\n",
      "minimum_value = 30260\n",
      "kluyveromyces 15130\n",
      "saccharomyces 30260\n",
      "candida 6052\n",
      "[30260, 30260, 30260]\n",
      "\n",
      "\n",
      "sporidiobolaceae\n",
      "['rhodotorula']\n",
      "{'rhodotorula': 1}\n",
      "minimum_value = 117800\n",
      "15000\n",
      "\n",
      "\n",
      "microascaceae\n",
      "['scedosporium']\n",
      "{'scedosporium': 1}\n",
      "minimum_value = 93722\n",
      "15000\n",
      "\n",
      "\n",
      "trichomonascaceae\n",
      "['blastobotrys' 'zygoascus']\n",
      "{'blastobotrys': 1, 'zygoascus': 1}\n",
      "minimum_value = 34002\n",
      "blastobotrys 15000\n",
      "zygoascus 15000\n",
      "[15000, 15000]\n",
      "\n",
      "\n",
      "debaryomycetaceae\n",
      "['debaryomyces' 'meyerozyma' 'yamadazyma']\n",
      "{'debaryomyces': 1, 'meyerozyma': 1, 'yamadazyma': 2}\n",
      "minimum_value = 18944\n",
      "debaryomyces 18944\n",
      "meyerozyma 18944\n",
      "yamadazyma 9472\n",
      "[18944, 18944, 18944]\n",
      "\n",
      "\n",
      "dipodascaceae\n",
      "['galactomyces' 'yarrowia' 'geotrichum']\n",
      "{'galactomyces': 1, 'yarrowia': 1, 'geotrichum': 1}\n",
      "minimum_value = 7804\n",
      "galactomyces 7804\n",
      "yarrowia 7804\n",
      "geotrichum 7804\n",
      "[7804, 7804, 7804]\n",
      "\n",
      "\n",
      "phaffomycetaceae\n",
      "['wickerhamomyces']\n",
      "{'wickerhamomyces': 1}\n",
      "minimum_value = 42588\n",
      "15000\n",
      "\n",
      "\n",
      "tremellaceae\n",
      "['cryptococcus']\n",
      "{'cryptococcus': 2}\n",
      "minimum_value = 23660\n",
      "15000\n",
      "\n",
      "\n",
      "diaporthaceae\n",
      "['diaporthe']\n",
      "{'diaporthe': 2}\n",
      "minimum_value = 27940\n",
      "15000\n",
      "\n",
      "\n",
      "dermateaceae\n",
      "['oculimacula']\n",
      "{'oculimacula': 2}\n",
      "minimum_value = 49480\n",
      "15000\n",
      "\n",
      "\n",
      "botryosphaeriaceae\n",
      "['dothiorella']\n",
      "{'dothiorella': 1}\n",
      "minimum_value = 44256\n",
      "15000\n",
      "\n",
      "\n",
      "quambalariaceae\n",
      "['quambalaria']\n",
      "{'quambalaria': 1}\n",
      "minimum_value = 46170\n",
      "15000\n",
      "\n",
      "\n",
      "xylariaceae\n",
      "['entoleuca']\n",
      "{'entoleuca': 1}\n",
      "minimum_value = 31356\n",
      "15000\n",
      "\n",
      "\n",
      "gnomoniaceae\n",
      "['asteroma']\n",
      "{'asteroma': 2}\n",
      "minimum_value = 31996\n",
      "15000\n",
      "\n",
      "\n",
      "herpotrichiellaceae\n",
      "['cladophialophora']\n",
      "{'cladophialophora': 1}\n",
      "minimum_value = 25916\n",
      "15000\n",
      "\n",
      "\n",
      "pichiaceae\n",
      "['pichia']\n",
      "{'pichia': 2}\n",
      "minimum_value = 26936\n",
      "15000\n",
      "order\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "pucciniales\n",
      "['pucciniaceae']\n",
      "{'pucciniaceae': 1}\n",
      "minimum_value = 113336\n",
      "15000\n",
      "\n",
      "\n",
      "capnodiales\n",
      "['mycosphaerellaceae']\n",
      "{'mycosphaerellaceae': 1}\n",
      "minimum_value = 133088\n",
      "15000\n",
      "\n",
      "\n",
      "pleosporales\n",
      "['pleosporaceae']\n",
      "{'pleosporaceae': 1}\n",
      "minimum_value = 90014\n",
      "15000\n",
      "\n",
      "\n",
      "hypocreales\n",
      "['nectriaceae']\n",
      "{'nectriaceae': 1}\n",
      "minimum_value = 123742\n",
      "15000\n",
      "\n",
      "\n",
      "pezizales\n",
      "['tuberaceae']\n",
      "{'tuberaceae': 1}\n",
      "minimum_value = 74232\n",
      "15000\n",
      "\n",
      "\n",
      "agaricales\n",
      "['cortinariaceae']\n",
      "{'cortinariaceae': 1}\n",
      "minimum_value = 117090\n",
      "15000\n",
      "\n",
      "\n",
      "eurotiales\n",
      "['trichocomaceae']\n",
      "{'trichocomaceae': 4}\n",
      "minimum_value = 39988\n",
      "15000\n",
      "\n",
      "\n",
      "saccharomycetales\n",
      "['metschnikowiaceae' 'saccharomycetaceae' 'trichomonascaceae'\n",
      " 'debaryomycetaceae' 'dipodascaceae' 'phaffomycetaceae' 'pichiaceae']\n",
      "{'metschnikowiaceae': 2, 'saccharomycetaceae': 8, 'trichomonascaceae': 2, 'debaryomycetaceae': 4, 'dipodascaceae': 3, 'phaffomycetaceae': 1, 'pichiaceae': 2}\n",
      "minimum_value = 42588\n",
      "metschnikowiaceae 21294\n",
      "saccharomycetaceae 5323\n",
      "trichomonascaceae 21294\n",
      "debaryomycetaceae 10647\n",
      "dipodascaceae 867\n",
      "phaffomycetaceae 2601\n",
      "pichiaceae 1300\n",
      "[42588, 42584, 42588, 42588, 2601, 2601, 2600]\n",
      "\n",
      "\n",
      "sporidiobolales\n",
      "['sporidiobolaceae']\n",
      "{'sporidiobolaceae': 1}\n",
      "minimum_value = 117800\n",
      "15000\n",
      "\n",
      "\n",
      "microascales\n",
      "['microascaceae']\n",
      "{'microascaceae': 1}\n",
      "minimum_value = 93722\n",
      "15000\n",
      "\n",
      "\n",
      "tremellales\n",
      "['tremellaceae']\n",
      "{'tremellaceae': 2}\n",
      "minimum_value = 23660\n",
      "15000\n",
      "\n",
      "\n",
      "diaporthales\n",
      "['diaporthaceae' 'gnomoniaceae']\n",
      "{'diaporthaceae': 2, 'gnomoniaceae': 2}\n",
      "minimum_value = 27940\n",
      "diaporthaceae 13970\n",
      "gnomoniaceae 13970\n",
      "[27940, 27940]\n",
      "\n",
      "\n",
      "helotiales\n",
      "['dermateaceae']\n",
      "{'dermateaceae': 2}\n",
      "minimum_value = 49480\n",
      "15000\n",
      "\n",
      "\n",
      "botryosphaeriales\n",
      "['botryosphaeriaceae']\n",
      "{'botryosphaeriaceae': 1}\n",
      "minimum_value = 44256\n",
      "15000\n",
      "\n",
      "\n",
      "microstromatales\n",
      "['quambalariaceae']\n",
      "{'quambalariaceae': 1}\n",
      "minimum_value = 46170\n",
      "15000\n",
      "\n",
      "\n",
      "xylariales\n",
      "['xylariaceae']\n",
      "{'xylariaceae': 1}\n",
      "minimum_value = 31356\n",
      "15000\n",
      "\n",
      "\n",
      "chaetothyriales\n",
      "['herpotrichiellaceae']\n",
      "{'herpotrichiellaceae': 1}\n",
      "minimum_value = 25916\n",
      "15000\n",
      "class\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "pucciniomycetes\n",
      "['pucciniales']\n",
      "{'pucciniales': 1}\n",
      "minimum_value = 113336\n",
      "15000\n",
      "\n",
      "\n",
      "dothideomycetes\n",
      "['capnodiales' 'pleosporales' 'botryosphaeriales']\n",
      "{'capnodiales': 1, 'pleosporales': 1, 'botryosphaeriales': 1}\n",
      "minimum_value = 44256\n",
      "capnodiales 15000\n",
      "pleosporales 15000\n",
      "botryosphaeriales 15000\n",
      "[15000, 15000, 15000]\n",
      "\n",
      "\n",
      "sordariomycetes\n",
      "['hypocreales' 'microascales' 'diaporthales' 'xylariales']\n",
      "{'hypocreales': 1, 'microascales': 1, 'diaporthales': 4, 'xylariales': 1}\n",
      "minimum_value = 31356\n",
      "hypocreales 31356\n",
      "microascales 31356\n",
      "diaporthales 7839\n",
      "xylariales 31356\n",
      "[31356, 31356, 31356, 31356]\n",
      "\n",
      "\n",
      "pezizomycetes\n",
      "['pezizales']\n",
      "{'pezizales': 1}\n",
      "minimum_value = 74232\n",
      "15000\n",
      "\n",
      "\n",
      "agaricomyctes\n",
      "['agaricales']\n",
      "{'agaricales': 1}\n",
      "minimum_value = 117090\n",
      "15000\n",
      "\n",
      "\n",
      "eurotiomycetes\n",
      "['eurotiales' 'chaetothyriales']\n",
      "{'eurotiales': 4, 'chaetothyriales': 1}\n",
      "minimum_value = 25916\n",
      "eurotiales 6479\n",
      "chaetothyriales 25916\n",
      "[25916, 25916]\n",
      "\n",
      "\n",
      "saccharomycetes\n",
      "['saccharomycetales']\n",
      "{'saccharomycetales': 22}\n",
      "minimum_value = 7804\n",
      "15000\n",
      "\n",
      "\n",
      "microbotryomycetes\n",
      "['sporidiobolales']\n",
      "{'sporidiobolales': 1}\n",
      "minimum_value = 117800\n",
      "15000\n",
      "\n",
      "\n",
      "tremellomycetes\n",
      "['tremellales']\n",
      "{'tremellales': 2}\n",
      "minimum_value = 23660\n",
      "15000\n",
      "\n",
      "\n",
      "leotiomycetes\n",
      "['helotiales']\n",
      "{'helotiales': 2}\n",
      "minimum_value = 49480\n",
      "15000\n",
      "\n",
      "\n",
      "exobasidiomycetes\n",
      "['microstromatales']\n",
      "{'microstromatales': 1}\n",
      "minimum_value = 46170\n",
      "15000\n",
      "phylum\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "basidiomycota\n",
      "['pucciniomycetes' 'agaricomyctes' 'microbotryomycetes' 'tremellomycetes'\n",
      " 'exobasidiomycetes']\n",
      "{'pucciniomycetes': 1, 'agaricomyctes': 1, 'microbotryomycetes': 1, 'tremellomycetes': 2, 'exobasidiomycetes': 1}\n",
      "minimum_value = 46170\n",
      "pucciniomycetes 30000\n",
      "agaricomyctes 30000\n",
      "microbotryomycetes 30000\n",
      "tremellomycetes 15000\n",
      "exobasidiomycetes 30000\n",
      "[30000, 30000, 30000, 30000, 30000]\n",
      "\n",
      "\n",
      "ascomycota\n",
      "['dothideomycetes' 'sordariomycetes' 'pezizomycetes' 'eurotiomycetes'\n",
      " 'saccharomycetes' 'leotiomycetes']\n",
      "{'dothideomycetes': 3, 'sordariomycetes': 7, 'pezizomycetes': 1, 'eurotiomycetes': 5, 'saccharomycetes': 22, 'leotiomycetes': 2}\n",
      "minimum_value = 74232\n",
      "dothideomycetes 24744\n",
      "sordariomycetes 10604\n",
      "pezizomycetes 74232\n",
      "eurotiomycetes 14846\n",
      "saccharomycetes 3374\n",
      "leotiomycetes 37116\n",
      "[74232, 74228, 74232, 74230, 74228, 74232]\n",
      "kingdom\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "fungi\n",
      "['basidiomycota' 'ascomycota']\n",
      "{'basidiomycota': 6, 'ascomycota': 40}\n",
      "minimum_value = 23660\n",
      "basidiomycota 3943\n",
      "ascomycota 591\n",
      "[23658, 23640]\n"
     ]
    }
   ],
   "source": [
    "for tax_rank in tax_ranks:\n",
    "    print(tax_rank + \"\\n\\n\")\n",
    "    for name in ref_df[tax_rank].unique():\n",
    "        total_class_reads = []\n",
    "        \n",
    "        print(\"\\n\\n\"+name)\n",
    "        indecies = ref_df[ref_df[tax_rank] == name].index\n",
    "        location = (ref_df.columns.get_loc(tax_rank)-1)\n",
    "        col_name = ref_df.columns[location]\n",
    "\n",
    "        classes = ref_df.iloc[indecies,location].unique()\n",
    "        print(classes)\n",
    "\n",
    "        count_dict = {}\n",
    "        for class_ in classes:\n",
    "            count_dict[class_] = sum(ref_df.iloc[indecies,location] == class_)\n",
    "        print(count_dict)\n",
    "\n",
    "        min_vals = []\n",
    "        for class_, n_class in count_dict.items():\n",
    "            if n_class == min(count_dict.values()):\n",
    "                min_vals.append(ref_df[ref_df.iloc[:,location] == class_]['# reads after length filtering'].min())\n",
    "        if min(min_vals) % 2 == 0:\n",
    "            minimum_value = min(min_vals)\n",
    "        else:\n",
    "            minimum_value = min(min_vals)-1\n",
    "        print(\"minimum_value =\", minimum_value)\n",
    "        \n",
    "        if len(count_dict) > 1:\n",
    "            max_reads = 0\n",
    "            for key, value in count_dict.items():\n",
    "                if value == max(count_dict.values()):\n",
    "                    max_reads = value*n_reads\n",
    "\n",
    "            if max_reads <= minimum_value:\n",
    "                minimum_value = max_reads\n",
    "                \n",
    "            for key, n_class in count_dict.items():\n",
    "                s_reads = int(minimum_value/n_class)\n",
    "                if ref_df[ref_df.loc[:,col_name]==key]['# reads after length filtering'].min() < s_reads:\n",
    "                    minimum_value = ref_df[ref_df.loc[:,col_name]==key]['# reads after length filtering'].min()/n_class\n",
    "                    s_reads = int(minimum_value/n_class)\n",
    "                print(key, s_reads)\n",
    "                total_class_reads.append(s_reads*n_class)\n",
    "            print(total_class_reads)\n",
    "        elif len(count_dict) == 1:\n",
    "            s_reads = n_reads\n",
    "            print(s_reads)\n",
    "#             print(\"no comparison for the rank\")\n",
    "#             exit()\n",
    "\n",
    "        # parse s_reads into numberfy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
