{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "import numpy as np\n",
    "import os\n",
    "import random"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Ideas of the program would be:\n",
    "\n",
    "- want to read in reference_dataframe.csv\n",
    "\n",
    "- taxonomic level e.g. genus=Candida -> species level comparision\n",
    "\n",
    "- path to location of data. Root of analysis/....."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "With arg parse\n",
    "\n",
    "ref_df_fn = args.ref_df_fn  \n",
    "data_root = args.data_root  \n",
    "tax_rank = args.tax_rank #use with flag e.g. -r  \n",
    "name = args.name #name of the rank to select from, use flag e.g. -n  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df_fn = '../../analysis/Stats/reference_dataframe.csv'\n",
    "ref_df_fn = os.path.abspath(ref_df_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "try:\n",
    "    os.path.exists(ref_df_fn)\n",
    "except:\n",
    "    print('Cannot find %s' % ref_df_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#to be able to handle anything from command lien\n",
    "tax_rank = 'family'\n",
    "name = 'Saccharomycetaceae'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_reads = 15000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "name = name.lower()\n",
    "tax_rank = tax_rank.lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df = pd.read_csv(ref_df_fn, index_col=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df[ref_df[\"# reads after length filtering\"] < n_reads].shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "try:\n",
    "    if ref_df[ref_df[\"# reads after length filtering\"] \\\n",
    "              < n_reads].shape[0] >0 :\n",
    "        print(\"These species need more reads.\")\n",
    "        print(ref_df[ref_df[\"# reads after length filtering\"] \\\n",
    "              < n_reads])\n",
    "        #exit()\n",
    "except:\n",
    "    print('Check %s to have the wanted column names' % ref_df_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#not neccessary pull from command line\n",
    "# name = \"Trichocomaceae\"\n",
    "data_root = '../../'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "indecies = ref_df[ref_df[tax_rank] == name].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "indecies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "SeqIO_dicts = {}\n",
    "for index in indecies:\n",
    "    fasta_path = os.path.join(data_root, ref_df.loc[index, 'path to length filtering'])\n",
    "    print(fasta_path)\n",
    "    try:\n",
    "        SeqIO_dicts[index] = SeqIO.to_dict(SeqIO.parse(fasta_path, 'fasta'))\n",
    "    except:\n",
    "        print('Check location of fasta files.')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def max_seq_len(SeqIO_dict):\n",
    "    \"\"\"\n",
    "    Fucntion takes a SeqIO_dict and returns the lenghts of the\n",
    "    longest sequence.\n",
    "    \"\"\"\n",
    "    total_lens = []\n",
    "    for key in SeqIO_dict.keys():\n",
    "        total_lens.append(len(SeqIO_dict[key].seq))\n",
    "    return max(total_lens)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def numberfi(SeqIO_dict, seq_len, nsubsample):\n",
    "    \"\"\"\n",
    "    Take SeqIO_dict and return SeqIO_dict were bases have been replaced\n",
    "    with numbers\n",
    "    ACGT- to 01234\n",
    "    Take the seq_len each sequence should have.\n",
    "    \"\"\"\n",
    "    num_dict = {}\n",
    "    \n",
    "    keys = list(SeqIO_dict.keys())\n",
    "    randkeys = random.sample(keys, k= nsubsample)\n",
    "    \n",
    "    \n",
    "    for key in randkeys:\n",
    "        seq = str(SeqIO_dict[key].seq).replace(\"A\",'0 ')\\\n",
    "        .replace(\"C\",'1 ').replace(\"G\",'2 ').replace(\"T\",'3 ')\n",
    "        seq_new = seq + '4 '*(seq_len -int(len(seq)/2))\n",
    "        num_dict[key] = list(map(int, seq_new.split(' ')[:-1]))\n",
    "    return num_dict\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "total_lens = []\n",
    "for key, value in SeqIO_dicts.items():\n",
    "    total_lens.append(max_seq_len(value))\n",
    "print(\"This is the maximum sequence length of all sampled sequences.\")\n",
    "print(total_lens)\n",
    "\n",
    "numSeqIO_dicts = {}\n",
    "max_len = max(total_lens)\n",
    "for key, value in SeqIO_dicts.items():\n",
    "    numSeqIO_dicts[key] = numberfi(value, max_len, s_reads)\n",
    "\n",
    "seq_list = []\n",
    "for index in indecies:\n",
    "    seq_list.append(np.array(list(numSeqIO_dicts[index].values())))\n",
    "seq_comb = np.concatenate(seq_list, axis = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_class = len(numSeqIO_dicts.keys())\n",
    "ids_comb = np.zeros( (n_reads*num_class,num_class) )\n",
    "for i in range(0, num_class):\n",
    "    ids_comb[i*n_reads:(i+1)*n_reads,i] = 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_class = 5\n",
    "ids_comb = np.zeros( (n_reads*num_class,num_class) )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(0, num_class):\n",
    "    ids_comb[i*n_reads:(i+1)*n_reads,i] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ids_comb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>species</th>\n",
       "      <th>genus</th>\n",
       "      <th>family</th>\n",
       "      <th>order</th>\n",
       "      <th>class</th>\n",
       "      <th>phylum</th>\n",
       "      <th>kingdom</th>\n",
       "      <th># raw reads</th>\n",
       "      <th># reads after homology filtering</th>\n",
       "      <th># reads after length filtering</th>\n",
       "      <th># reads for use</th>\n",
       "      <th>path to raw reads</th>\n",
       "      <th>path to homology filtering</th>\n",
       "      <th>path to length filtering</th>\n",
       "      <th>path for use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>20171103_FAH15473/barcode01</td>\n",
       "      <td>striiformis-tritici</td>\n",
       "      <td>puccinia</td>\n",
       "      <td>pucciniaceae</td>\n",
       "      <td>pucciniales</td>\n",
       "      <td>pucciniomycetes</td>\n",
       "      <td>basidiomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>272465</td>\n",
       "      <td>122080</td>\n",
       "      <td>113337</td>\n",
       "      <td>113337.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>20171103_FAH15473/barcode02</td>\n",
       "      <td>tritici</td>\n",
       "      <td>zymoseptoria</td>\n",
       "      <td>mycosphaerellaceae</td>\n",
       "      <td>capnodiales</td>\n",
       "      <td>dothideomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>413127</td>\n",
       "      <td>143363</td>\n",
       "      <td>133089</td>\n",
       "      <td>133089.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>20171103_FAH15473/barcode03</td>\n",
       "      <td>tritici-repentis</td>\n",
       "      <td>pyrenophora</td>\n",
       "      <td>pleosporaceae</td>\n",
       "      <td>pleosporales</td>\n",
       "      <td>dothideomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>260896</td>\n",
       "      <td>97584</td>\n",
       "      <td>90015</td>\n",
       "      <td>90015.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>20171103_FAH15473/barcode04</td>\n",
       "      <td>oxysporum</td>\n",
       "      <td>fusarium</td>\n",
       "      <td>nectriaceae</td>\n",
       "      <td>hypocreales</td>\n",
       "      <td>sordariomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>382450</td>\n",
       "      <td>131411</td>\n",
       "      <td>123742</td>\n",
       "      <td>123742.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>20171103_FAH15473/barcode05</td>\n",
       "      <td>brumale</td>\n",
       "      <td>tuber</td>\n",
       "      <td>tuberaceae</td>\n",
       "      <td>pezizales</td>\n",
       "      <td>pezizomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>275035</td>\n",
       "      <td>80614</td>\n",
       "      <td>74232</td>\n",
       "      <td>74232.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    Unnamed: 0              species         genus  \\\n",
       "0  20171103_FAH15473/barcode01  striiformis-tritici      puccinia   \n",
       "1  20171103_FAH15473/barcode02              tritici  zymoseptoria   \n",
       "2  20171103_FAH15473/barcode03     tritici-repentis   pyrenophora   \n",
       "3  20171103_FAH15473/barcode04            oxysporum      fusarium   \n",
       "4  20171103_FAH15473/barcode05              brumale         tuber   \n",
       "\n",
       "               family         order            class         phylum kingdom  \\\n",
       "0        pucciniaceae   pucciniales  pucciniomycetes  basidiomycota   fungi   \n",
       "1  mycosphaerellaceae   capnodiales  dothideomycetes     ascomycota   fungi   \n",
       "2       pleosporaceae  pleosporales  dothideomycetes     ascomycota   fungi   \n",
       "3         nectriaceae   hypocreales  sordariomycetes     ascomycota   fungi   \n",
       "4          tuberaceae     pezizales    pezizomycetes     ascomycota   fungi   \n",
       "\n",
       "   # raw reads  # reads after homology filtering  \\\n",
       "0       272465                            122080   \n",
       "1       413127                            143363   \n",
       "2       260896                             97584   \n",
       "3       382450                            131411   \n",
       "4       275035                             80614   \n",
       "\n",
       "   # reads after length filtering  # reads for use  \\\n",
       "0                          113337         113337.0   \n",
       "1                          133089         133089.0   \n",
       "2                           90015          90015.0   \n",
       "3                          123742         123742.0   \n",
       "4                           74232          74232.0   \n",
       "\n",
       "                                   path to raw reads  \\\n",
       "0  analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "1  analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "2  analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "3  analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "4  analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "\n",
       "                          path to homology filtering  \\\n",
       "0  analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "1  analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "2  analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "3  analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "4  analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "\n",
       "                            path to length filtering  \\\n",
       "0  analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "1  analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "2  analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "3  analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "4  analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "\n",
       "                                        path for use  \n",
       "0  analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "1  analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "2  analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "3  analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "4  analysis/Length_Filtered/20171103_FAH15473/bar...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'numpy.int64'>\n"
     ]
    }
   ],
   "source": [
    "print(type(ref_df.loc[0,'# reads after length filtering']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'numpy.float64'>\n"
     ]
    }
   ],
   "source": [
    "print(type(ref_df.loc[0,'# reads for use']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df[tax_rank] == 'Candida'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "location = (ref_df.columns.get_loc(tax_rank)-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "classes = ref_df.iloc[indecies,location].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_dict = {}\n",
    "for item in classes:\n",
    "    count_dict[item] = sum(ref_df.iloc[:,location] == item)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "classes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for item in count_dict.keys():\n",
    "    print(n_reads/count_dict[item])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# define n_reads\n",
    "\n",
    "min_vals = []\n",
    "for key, value in count_dict.items():\n",
    "    if value == min(count_dict.values()):\n",
    "        min_vals.append(ref_df[ref_df.iloc[:,location] == item]['# reads after length filtering'].min())\n",
    "min(min_vals)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "max_reads = 0\n",
    "for key, value in count_dict.items():\n",
    "    if value == max(count_dict.values()):\n",
    "        max_reads = value*n_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "max_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "min_vals = [44999]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# is max_reads < min_reads?\n",
    "# if yes, use max_reads to subsample as base value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if max_reads <= min(min_vals):\n",
    "    for key, value in count_dict.items():\n",
    "        s_reads = int(max_reads/value)\n",
    "        print(key, s_reads)\n",
    "else:\n",
    "    for key, value in count_dict.items():\n",
    "        s_reads = int(min(min_vals)/value)\n",
    "        print(key, s_reads)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "location = (ref_df.columns.get_loc(tax_rank)-1)\n",
    "col_name = ref_df.columns[location]\n",
    "\n",
    "classes = ref_df.iloc[indecies,location].unique()\n",
    "\n",
    "count_dict = {}\n",
    "for item in classes:\n",
    "    count_dict[item] = sum(ref_df.iloc[:,location] == item)\n",
    "\n",
    "min_vals = []\n",
    "for key, value in count_dict.items():\n",
    "    if value == min(count_dict.values()):\n",
    "        min_vals.append(ref_df[ref_df.iloc[:,location] == item]['# reads after length filtering'].min())\n",
    "minimum_value = min(min_vals)\n",
    "\n",
    "max_reads = 0\n",
    "for key, value in count_dict.items():\n",
    "    if value == max(count_dict.values()):\n",
    "        max_reads = value*n_reads\n",
    "\n",
    "if max_reads <= minimum_value:\n",
    "    minimum_value = max_reads\n",
    "for key, n_class in count_dict.items():\n",
    "    s_reads = int(minimum_value/n_class)\n",
    "    if ref_df[ref_df.loc[:,col_name]==key]['# reads after length filtering'].min() < s_reads:\n",
    "        minimum_value = ref_df[ref_df.loc[:,col_name]==key]['# reads after length filtering'].min()/n_class\n",
    "        s_reads = int(minimum_value/n_class)\n",
    "    print(key, s_reads)\n",
    "    \n",
    "# parse s_reads into numberfy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tax_ranks = ref_df.columns[2:-6]\n",
    "print(tax_ranks)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for tax_rank in tax_ranks:\n",
    "    print(tax_rank + \"\\n\\n\")\n",
    "    for name in ref_df[tax_rank].unique():\n",
    "        total_class_reads = []\n",
    "        \n",
    "        print(\"\\n\\n\"+name)\n",
    "        indecies = ref_df[ref_df[tax_rank] == name].index\n",
    "        location = (ref_df.columns.get_loc(tax_rank)-1)\n",
    "        col_name = ref_df.columns[location]\n",
    "\n",
    "        classes = ref_df.iloc[indecies,location].unique()\n",
    "        print(classes)\n",
    "\n",
    "        count_dict = {}\n",
    "        for class_ in classes:\n",
    "            count_dict[class_] = sum(ref_df.iloc[indecies,location] == class_)\n",
    "        print(count_dict)\n",
    "\n",
    "        min_vals = []\n",
    "        for class_, n_class in count_dict.items():\n",
    "            if n_class == min(count_dict.values()):\n",
    "                min_vals.append(ref_df[ref_df.iloc[:,location] == class_]['# reads after length filtering'].min())\n",
    "        if min(min_vals) % 2 == 0:\n",
    "            minimum_value = min(min_vals)\n",
    "        else:\n",
    "            minimum_value = min(min_vals)-1\n",
    "        print(\"minimum_value =\", minimum_value)\n",
    "        \n",
    "        if len(count_dict) > 1:\n",
    "            max_reads = 0\n",
    "            for key, value in count_dict.items():\n",
    "                if value == max(count_dict.values()):\n",
    "                    max_reads = value*n_reads\n",
    "\n",
    "            if max_reads <= minimum_value:\n",
    "                minimum_value = max_reads\n",
    "                \n",
    "            for key, n_class in count_dict.items():\n",
    "                s_reads = int(minimum_value/n_class)\n",
    "                if ref_df[ref_df.loc[:,col_name]==key]['# reads after length filtering'].min() < s_reads:\n",
    "                    minimum_value = ref_df[ref_df.loc[:,col_name]==key]['# reads after length filtering'].min()/n_class\n",
    "                    s_reads = int(minimum_value/n_class)\n",
    "                print(key, s_reads)\n",
    "                total_class_reads.append(s_reads*n_class)\n",
    "            print(total_class_reads)\n",
    "        elif len(count_dict) == 1:\n",
    "            s_reads = n_reads\n",
    "            print(s_reads)\n",
    "#             print(\"no comparison for the rank\")\n",
    "#             exit()\n",
    "\n",
    "        # parse s_reads into numberfy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>species</th>\n",
       "      <th>genus</th>\n",
       "      <th>family</th>\n",
       "      <th>order</th>\n",
       "      <th>class</th>\n",
       "      <th>phylum</th>\n",
       "      <th>kingdom</th>\n",
       "      <th># raw reads</th>\n",
       "      <th># reads after homology filtering</th>\n",
       "      <th># reads after length filtering</th>\n",
       "      <th># for use</th>\n",
       "      <th>path to raw reads</th>\n",
       "      <th>path to homology filtering</th>\n",
       "      <th>path to length filtering</th>\n",
       "      <th>path for use</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>20171103_FAH15473/barcode01</td>\n",
       "      <td>striiformis-tritici</td>\n",
       "      <td>puccinia</td>\n",
       "      <td>pucciniaceae</td>\n",
       "      <td>pucciniales</td>\n",
       "      <td>pucciniomycetes</td>\n",
       "      <td>basidiomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>272465</td>\n",
       "      <td>122080</td>\n",
       "      <td>113337</td>\n",
       "      <td>113337.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>20171103_FAH15473/barcode02</td>\n",
       "      <td>tritici</td>\n",
       "      <td>zymoseptoria</td>\n",
       "      <td>mycosphaerellaceae</td>\n",
       "      <td>capnodiales</td>\n",
       "      <td>dothideomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>413127</td>\n",
       "      <td>143363</td>\n",
       "      <td>133089</td>\n",
       "      <td>133089.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>20171103_FAH15473/barcode03</td>\n",
       "      <td>tritici-repentis</td>\n",
       "      <td>pyrenophora</td>\n",
       "      <td>pleosporaceae</td>\n",
       "      <td>pleosporales</td>\n",
       "      <td>dothideomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>260896</td>\n",
       "      <td>97584</td>\n",
       "      <td>90015</td>\n",
       "      <td>90015.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>20171103_FAH15473/barcode04</td>\n",
       "      <td>oxysporum</td>\n",
       "      <td>fusarium</td>\n",
       "      <td>nectriaceae</td>\n",
       "      <td>hypocreales</td>\n",
       "      <td>sordariomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>382450</td>\n",
       "      <td>131411</td>\n",
       "      <td>123742</td>\n",
       "      <td>123742.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>20171103_FAH15473/barcode05</td>\n",
       "      <td>brumale</td>\n",
       "      <td>tuber</td>\n",
       "      <td>tuberaceae</td>\n",
       "      <td>pezizales</td>\n",
       "      <td>pezizomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>275035</td>\n",
       "      <td>80614</td>\n",
       "      <td>74232</td>\n",
       "      <td>74232.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>20171103_FAH15473/barcode06</td>\n",
       "      <td>globuliformis</td>\n",
       "      <td>cortinarius</td>\n",
       "      <td>cortinariaceae</td>\n",
       "      <td>agaricales</td>\n",
       "      <td>agaricomyctes</td>\n",
       "      <td>basidiomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>347423</td>\n",
       "      <td>128993</td>\n",
       "      <td>117090</td>\n",
       "      <td>117090.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>20171103_FAH15473/barcode07</td>\n",
       "      <td>niger</td>\n",
       "      <td>aspergillus</td>\n",
       "      <td>trichocomaceae</td>\n",
       "      <td>eurotiales</td>\n",
       "      <td>eurotiomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>171615</td>\n",
       "      <td>65065</td>\n",
       "      <td>59406</td>\n",
       "      <td>59406.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>20171103_FAH15473/barcode08</td>\n",
       "      <td>lusitaniae</td>\n",
       "      <td>clavispora</td>\n",
       "      <td>metschnikowiaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>352768</td>\n",
       "      <td>141856</td>\n",
       "      <td>131936</td>\n",
       "      <td>131936.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>20171103_FAH15473/barcode09</td>\n",
       "      <td>unidentified</td>\n",
       "      <td>kluyveromyces</td>\n",
       "      <td>saccharomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>370154</td>\n",
       "      <td>165113</td>\n",
       "      <td>152736</td>\n",
       "      <td>152736.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>20171103_FAH15473/barcode10</td>\n",
       "      <td>chrysogenum</td>\n",
       "      <td>penicillium</td>\n",
       "      <td>trichocomaceae</td>\n",
       "      <td>eurotiales</td>\n",
       "      <td>eurotiomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>192173</td>\n",
       "      <td>78105</td>\n",
       "      <td>72307</td>\n",
       "      <td>72307.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>10</td>\n",
       "      <td>20171103_FAH15473/barcode11</td>\n",
       "      <td>mucilaginosa</td>\n",
       "      <td>rhodotorula</td>\n",
       "      <td>sporidiobolaceae</td>\n",
       "      <td>sporidiobolales</td>\n",
       "      <td>microbotryomycetes</td>\n",
       "      <td>basidiomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>318405</td>\n",
       "      <td>127801</td>\n",
       "      <td>117801</td>\n",
       "      <td>117801.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>11</td>\n",
       "      <td>20171103_FAH15473/barcode12</td>\n",
       "      <td>boydii</td>\n",
       "      <td>scedosporium</td>\n",
       "      <td>microascaceae</td>\n",
       "      <td>microascales</td>\n",
       "      <td>sordariomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>331947</td>\n",
       "      <td>102481</td>\n",
       "      <td>93723</td>\n",
       "      <td>93723.0</td>\n",
       "      <td>analysis/Concatenated/20171103_FAH15473/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171103_FAH15473/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171103_FAH15473/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>12</td>\n",
       "      <td>20171207_FAH18654/barcode01</td>\n",
       "      <td>proliferans</td>\n",
       "      <td>blastobotrys</td>\n",
       "      <td>trichomonascaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>133835</td>\n",
       "      <td>39631</td>\n",
       "      <td>37315</td>\n",
       "      <td>37315.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>13</td>\n",
       "      <td>20171207_FAH18654/barcode02</td>\n",
       "      <td>unidentified</td>\n",
       "      <td>debaryomyces</td>\n",
       "      <td>debaryomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>174974</td>\n",
       "      <td>35837</td>\n",
       "      <td>33499</td>\n",
       "      <td>33499.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>14</td>\n",
       "      <td>20171207_FAH18654/barcode03</td>\n",
       "      <td>geotrichum</td>\n",
       "      <td>galactomyces</td>\n",
       "      <td>dipodascaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>152933</td>\n",
       "      <td>8485</td>\n",
       "      <td>7805</td>\n",
       "      <td>16587.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>15</td>\n",
       "      <td>20171207_FAH18654/barcode04</td>\n",
       "      <td>ohmeri</td>\n",
       "      <td>kodamaea</td>\n",
       "      <td>metschnikowiaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>111257</td>\n",
       "      <td>38931</td>\n",
       "      <td>36478</td>\n",
       "      <td>36478.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>16</td>\n",
       "      <td>20171207_FAH18654/barcode05</td>\n",
       "      <td>guillermondii</td>\n",
       "      <td>meyerozyma</td>\n",
       "      <td>debaryomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>211853</td>\n",
       "      <td>20333</td>\n",
       "      <td>18944</td>\n",
       "      <td>18944.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>17</td>\n",
       "      <td>20171207_FAH18654/barcode06</td>\n",
       "      <td>anomalus</td>\n",
       "      <td>wickerhamomyces</td>\n",
       "      <td>phaffomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>193187</td>\n",
       "      <td>45720</td>\n",
       "      <td>42589</td>\n",
       "      <td>42589.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>18</td>\n",
       "      <td>20171207_FAH18654/barcode07</td>\n",
       "      <td>mexicana</td>\n",
       "      <td>yamadazyma</td>\n",
       "      <td>debaryomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>179240</td>\n",
       "      <td>45093</td>\n",
       "      <td>42369</td>\n",
       "      <td>42369.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>19</td>\n",
       "      <td>20171207_FAH18654/barcode08</td>\n",
       "      <td>scolyti</td>\n",
       "      <td>yamadazyma</td>\n",
       "      <td>debaryomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>136650</td>\n",
       "      <td>37159</td>\n",
       "      <td>34841</td>\n",
       "      <td>34841.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>20</td>\n",
       "      <td>20171207_FAH18654/barcode09</td>\n",
       "      <td>lipolytica</td>\n",
       "      <td>yarrowia</td>\n",
       "      <td>dipodascaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>141238</td>\n",
       "      <td>35950</td>\n",
       "      <td>33873</td>\n",
       "      <td>33873.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>21</td>\n",
       "      <td>20171207_FAH18654/barcode11</td>\n",
       "      <td>hellenicus</td>\n",
       "      <td>zygoascus</td>\n",
       "      <td>trichomonascaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>229073</td>\n",
       "      <td>36666</td>\n",
       "      <td>34002</td>\n",
       "      <td>34002.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>22</td>\n",
       "      <td>20171207_FAH18654/barcode12</td>\n",
       "      <td>flavus</td>\n",
       "      <td>aspergillus</td>\n",
       "      <td>trichocomaceae</td>\n",
       "      <td>eurotiales</td>\n",
       "      <td>eurotiomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>125014</td>\n",
       "      <td>54061</td>\n",
       "      <td>51340</td>\n",
       "      <td>51340.0</td>\n",
       "      <td>analysis/Concatenated/20171207_FAH18654/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171207_FAH18654/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171207_FAH18654/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>23</td>\n",
       "      <td>20171212_FAH18688/barcode01</td>\n",
       "      <td>zero</td>\n",
       "      <td>cryptococcus</td>\n",
       "      <td>tremellaceae</td>\n",
       "      <td>tremellales</td>\n",
       "      <td>tremellomycetes</td>\n",
       "      <td>basidiomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>167818</td>\n",
       "      <td>42373</td>\n",
       "      <td>39235</td>\n",
       "      <td>39235.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>24</td>\n",
       "      <td>20171212_FAH18688/barcode02</td>\n",
       "      <td>unidentified</td>\n",
       "      <td>aspergillus</td>\n",
       "      <td>trichocomaceae</td>\n",
       "      <td>eurotiales</td>\n",
       "      <td>eurotiomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>249468</td>\n",
       "      <td>42899</td>\n",
       "      <td>39988</td>\n",
       "      <td>39988.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>25</td>\n",
       "      <td>20171212_FAH18688/barcode03</td>\n",
       "      <td>ccl067</td>\n",
       "      <td>diaporthe</td>\n",
       "      <td>diaporthaceae</td>\n",
       "      <td>diaporthales</td>\n",
       "      <td>sordariomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>172601</td>\n",
       "      <td>32847</td>\n",
       "      <td>30504</td>\n",
       "      <td>30504.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>26</td>\n",
       "      <td>20171212_FAH18688/barcode04</td>\n",
       "      <td>unidentified</td>\n",
       "      <td>diaporthe</td>\n",
       "      <td>diaporthaceae</td>\n",
       "      <td>diaporthales</td>\n",
       "      <td>sordariomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>198500</td>\n",
       "      <td>29833</td>\n",
       "      <td>27941</td>\n",
       "      <td>27941.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>27</td>\n",
       "      <td>20171212_FAH18688/barcode05</td>\n",
       "      <td>yallundae-ccl031</td>\n",
       "      <td>oculimacula</td>\n",
       "      <td>dermateaceae</td>\n",
       "      <td>helotiales</td>\n",
       "      <td>leotiomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>213143</td>\n",
       "      <td>52944</td>\n",
       "      <td>49481</td>\n",
       "      <td>49481.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>28</td>\n",
       "      <td>20171212_FAH18688/barcode06</td>\n",
       "      <td>yallundae-ccl029</td>\n",
       "      <td>oculimacula</td>\n",
       "      <td>dermateaceae</td>\n",
       "      <td>helotiales</td>\n",
       "      <td>leotiomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>223186</td>\n",
       "      <td>59651</td>\n",
       "      <td>55589</td>\n",
       "      <td>55589.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>29</td>\n",
       "      <td>20171212_FAH18688/barcode07</td>\n",
       "      <td>vidmadera</td>\n",
       "      <td>dothiorella</td>\n",
       "      <td>botryosphaeriaceae</td>\n",
       "      <td>botryosphaeriales</td>\n",
       "      <td>dothideomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>204777</td>\n",
       "      <td>47318</td>\n",
       "      <td>44257</td>\n",
       "      <td>44257.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>30</td>\n",
       "      <td>20171212_FAH18688/barcode08</td>\n",
       "      <td>cyanescens</td>\n",
       "      <td>quambalaria</td>\n",
       "      <td>quambalariaceae</td>\n",
       "      <td>microstromatales</td>\n",
       "      <td>exobasidiomycetes</td>\n",
       "      <td>basidiomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>205404</td>\n",
       "      <td>49780</td>\n",
       "      <td>46171</td>\n",
       "      <td>46171.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>31</td>\n",
       "      <td>20171212_FAH18688/barcode09</td>\n",
       "      <td>unidentified</td>\n",
       "      <td>entoleuca</td>\n",
       "      <td>xylariaceae</td>\n",
       "      <td>xylariales</td>\n",
       "      <td>sordariomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>155158</td>\n",
       "      <td>33941</td>\n",
       "      <td>31356</td>\n",
       "      <td>31356.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>32</td>\n",
       "      <td>20171212_FAH18688/barcode11</td>\n",
       "      <td>ccl060</td>\n",
       "      <td>asteroma</td>\n",
       "      <td>gnomoniaceae</td>\n",
       "      <td>diaporthales</td>\n",
       "      <td>sordariomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>206161</td>\n",
       "      <td>42329</td>\n",
       "      <td>39836</td>\n",
       "      <td>39836.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>33</td>\n",
       "      <td>20171212_FAH18688/barcode12</td>\n",
       "      <td>ccl068</td>\n",
       "      <td>asteroma</td>\n",
       "      <td>gnomoniaceae</td>\n",
       "      <td>diaporthales</td>\n",
       "      <td>sordariomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>188353</td>\n",
       "      <td>33438</td>\n",
       "      <td>31996</td>\n",
       "      <td>31996.0</td>\n",
       "      <td>analysis/Concatenated/20171212_FAH18688/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20171212_FAH18688/b...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20171212_FAH18688/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>34</td>\n",
       "      <td>20180108_FAH18647/barcode01</td>\n",
       "      <td>cerevisiae</td>\n",
       "      <td>saccharomyces</td>\n",
       "      <td>saccharomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>96837</td>\n",
       "      <td>33025</td>\n",
       "      <td>30260</td>\n",
       "      <td>30260.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>35</td>\n",
       "      <td>20180108_FAH18647/barcode02</td>\n",
       "      <td>unidentified</td>\n",
       "      <td>cladophialophora</td>\n",
       "      <td>herpotrichiellaceae</td>\n",
       "      <td>chaetothyriales</td>\n",
       "      <td>eurotiomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>115477</td>\n",
       "      <td>28063</td>\n",
       "      <td>25917</td>\n",
       "      <td>25917.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>36</td>\n",
       "      <td>20180108_FAH18647/barcode03</td>\n",
       "      <td>albicans</td>\n",
       "      <td>candida</td>\n",
       "      <td>saccharomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>91031</td>\n",
       "      <td>26114</td>\n",
       "      <td>25597</td>\n",
       "      <td>25597.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>37</td>\n",
       "      <td>20180108_FAH18647/barcode04</td>\n",
       "      <td>metapsilosis</td>\n",
       "      <td>candida</td>\n",
       "      <td>saccharomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>127633</td>\n",
       "      <td>40257</td>\n",
       "      <td>37426</td>\n",
       "      <td>37426.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>38</td>\n",
       "      <td>20180108_FAH18647/barcode05</td>\n",
       "      <td>orthopsilosis</td>\n",
       "      <td>candida</td>\n",
       "      <td>saccharomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>104214</td>\n",
       "      <td>32490</td>\n",
       "      <td>29765</td>\n",
       "      <td>29765.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>39</td>\n",
       "      <td>20180108_FAH18647/barcode06</td>\n",
       "      <td>parapsilosis</td>\n",
       "      <td>candida</td>\n",
       "      <td>saccharomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>135720</td>\n",
       "      <td>45958</td>\n",
       "      <td>42338</td>\n",
       "      <td>42338.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>40</td>\n",
       "      <td>20180108_FAH18647/barcode09</td>\n",
       "      <td>unidentified</td>\n",
       "      <td>candida</td>\n",
       "      <td>saccharomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>109905</td>\n",
       "      <td>38085</td>\n",
       "      <td>35120</td>\n",
       "      <td>35120.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>41</td>\n",
       "      <td>20180108_FAH18647/barcode10</td>\n",
       "      <td>marxianus</td>\n",
       "      <td>kluyveromyces</td>\n",
       "      <td>saccharomycetaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>115282</td>\n",
       "      <td>31150</td>\n",
       "      <td>28382</td>\n",
       "      <td>28382.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>20180108_FAH18647/barcode11</td>\n",
       "      <td>kudriavzevii</td>\n",
       "      <td>pichia</td>\n",
       "      <td>pichiaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>122601</td>\n",
       "      <td>35604</td>\n",
       "      <td>33244</td>\n",
       "      <td>33244.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43</td>\n",
       "      <td>20180108_FAH18647/barcode12</td>\n",
       "      <td>membranifaciens</td>\n",
       "      <td>pichia</td>\n",
       "      <td>pichiaceae</td>\n",
       "      <td>saccharomycetales</td>\n",
       "      <td>saccharomycetes</td>\n",
       "      <td>ascomycota</td>\n",
       "      <td>fungi</td>\n",
       "      <td>104844</td>\n",
       "      <td>29540</td>\n",
       "      <td>26937</td>\n",
       "      <td>26937.0</td>\n",
       "      <td>analysis/Concatenated/20180108_FAH18647/barcod...</td>\n",
       "      <td>analysis/Python_Processing/20180108_FAH18647/b...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "      <td>analysis/Length_Filtered/20180108_FAH18647/bar...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Unnamed: 0              species             genus  \\\n",
       "0   20171103_FAH15473/barcode01  striiformis-tritici          puccinia   \n",
       "1   20171103_FAH15473/barcode02              tritici      zymoseptoria   \n",
       "2   20171103_FAH15473/barcode03     tritici-repentis       pyrenophora   \n",
       "3   20171103_FAH15473/barcode04            oxysporum          fusarium   \n",
       "4   20171103_FAH15473/barcode05              brumale             tuber   \n",
       "5   20171103_FAH15473/barcode06        globuliformis       cortinarius   \n",
       "6   20171103_FAH15473/barcode07                niger       aspergillus   \n",
       "7   20171103_FAH15473/barcode08           lusitaniae        clavispora   \n",
       "8   20171103_FAH15473/barcode09         unidentified     kluyveromyces   \n",
       "9   20171103_FAH15473/barcode10          chrysogenum       penicillium   \n",
       "10  20171103_FAH15473/barcode11         mucilaginosa       rhodotorula   \n",
       "11  20171103_FAH15473/barcode12               boydii      scedosporium   \n",
       "12  20171207_FAH18654/barcode01          proliferans      blastobotrys   \n",
       "13  20171207_FAH18654/barcode02         unidentified      debaryomyces   \n",
       "14  20171207_FAH18654/barcode03           geotrichum      galactomyces   \n",
       "15  20171207_FAH18654/barcode04               ohmeri          kodamaea   \n",
       "16  20171207_FAH18654/barcode05        guillermondii        meyerozyma   \n",
       "17  20171207_FAH18654/barcode06             anomalus   wickerhamomyces   \n",
       "18  20171207_FAH18654/barcode07             mexicana        yamadazyma   \n",
       "19  20171207_FAH18654/barcode08              scolyti        yamadazyma   \n",
       "20  20171207_FAH18654/barcode09           lipolytica          yarrowia   \n",
       "21  20171207_FAH18654/barcode11           hellenicus         zygoascus   \n",
       "22  20171207_FAH18654/barcode12               flavus       aspergillus   \n",
       "23  20171212_FAH18688/barcode01                 zero      cryptococcus   \n",
       "24  20171212_FAH18688/barcode02         unidentified       aspergillus   \n",
       "25  20171212_FAH18688/barcode03               ccl067         diaporthe   \n",
       "26  20171212_FAH18688/barcode04         unidentified         diaporthe   \n",
       "27  20171212_FAH18688/barcode05     yallundae-ccl031       oculimacula   \n",
       "28  20171212_FAH18688/barcode06     yallundae-ccl029       oculimacula   \n",
       "29  20171212_FAH18688/barcode07            vidmadera       dothiorella   \n",
       "30  20171212_FAH18688/barcode08           cyanescens       quambalaria   \n",
       "31  20171212_FAH18688/barcode09         unidentified         entoleuca   \n",
       "32  20171212_FAH18688/barcode11               ccl060          asteroma   \n",
       "33  20171212_FAH18688/barcode12               ccl068          asteroma   \n",
       "34  20180108_FAH18647/barcode01           cerevisiae     saccharomyces   \n",
       "35  20180108_FAH18647/barcode02         unidentified  cladophialophora   \n",
       "36  20180108_FAH18647/barcode03             albicans           candida   \n",
       "37  20180108_FAH18647/barcode04         metapsilosis           candida   \n",
       "38  20180108_FAH18647/barcode05        orthopsilosis           candida   \n",
       "39  20180108_FAH18647/barcode06         parapsilosis           candida   \n",
       "40  20180108_FAH18647/barcode09         unidentified           candida   \n",
       "41  20180108_FAH18647/barcode10            marxianus     kluyveromyces   \n",
       "42  20180108_FAH18647/barcode11         kudriavzevii            pichia   \n",
       "43  20180108_FAH18647/barcode12      membranifaciens            pichia   \n",
       "\n",
       "                 family              order               class         phylum  \\\n",
       "0          pucciniaceae        pucciniales     pucciniomycetes  basidiomycota   \n",
       "1    mycosphaerellaceae        capnodiales     dothideomycetes     ascomycota   \n",
       "2         pleosporaceae       pleosporales     dothideomycetes     ascomycota   \n",
       "3           nectriaceae        hypocreales     sordariomycetes     ascomycota   \n",
       "4            tuberaceae          pezizales       pezizomycetes     ascomycota   \n",
       "5        cortinariaceae         agaricales       agaricomyctes  basidiomycota   \n",
       "6        trichocomaceae         eurotiales      eurotiomycetes     ascomycota   \n",
       "7     metschnikowiaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "8    saccharomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "9        trichocomaceae         eurotiales      eurotiomycetes     ascomycota   \n",
       "10     sporidiobolaceae    sporidiobolales  microbotryomycetes  basidiomycota   \n",
       "11        microascaceae       microascales     sordariomycetes     ascomycota   \n",
       "12    trichomonascaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "13    debaryomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "14        dipodascaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "15    metschnikowiaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "16    debaryomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "17     phaffomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "18    debaryomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "19    debaryomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "20        dipodascaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "21    trichomonascaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "22       trichocomaceae         eurotiales      eurotiomycetes     ascomycota   \n",
       "23         tremellaceae        tremellales     tremellomycetes  basidiomycota   \n",
       "24       trichocomaceae         eurotiales      eurotiomycetes     ascomycota   \n",
       "25        diaporthaceae       diaporthales     sordariomycetes     ascomycota   \n",
       "26        diaporthaceae       diaporthales     sordariomycetes     ascomycota   \n",
       "27         dermateaceae         helotiales       leotiomycetes     ascomycota   \n",
       "28         dermateaceae         helotiales       leotiomycetes     ascomycota   \n",
       "29   botryosphaeriaceae  botryosphaeriales     dothideomycetes     ascomycota   \n",
       "30      quambalariaceae   microstromatales   exobasidiomycetes  basidiomycota   \n",
       "31          xylariaceae         xylariales     sordariomycetes     ascomycota   \n",
       "32         gnomoniaceae       diaporthales     sordariomycetes     ascomycota   \n",
       "33         gnomoniaceae       diaporthales     sordariomycetes     ascomycota   \n",
       "34   saccharomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "35  herpotrichiellaceae    chaetothyriales      eurotiomycetes     ascomycota   \n",
       "36   saccharomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "37   saccharomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "38   saccharomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "39   saccharomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "40   saccharomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "41   saccharomycetaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "42           pichiaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "43           pichiaceae  saccharomycetales     saccharomycetes     ascomycota   \n",
       "\n",
       "   kingdom  # raw reads  # reads after homology filtering  \\\n",
       "0    fungi       272465                            122080   \n",
       "1    fungi       413127                            143363   \n",
       "2    fungi       260896                             97584   \n",
       "3    fungi       382450                            131411   \n",
       "4    fungi       275035                             80614   \n",
       "5    fungi       347423                            128993   \n",
       "6    fungi       171615                             65065   \n",
       "7    fungi       352768                            141856   \n",
       "8    fungi       370154                            165113   \n",
       "9    fungi       192173                             78105   \n",
       "10   fungi       318405                            127801   \n",
       "11   fungi       331947                            102481   \n",
       "12   fungi       133835                             39631   \n",
       "13   fungi       174974                             35837   \n",
       "14   fungi       152933                              8485   \n",
       "15   fungi       111257                             38931   \n",
       "16   fungi       211853                             20333   \n",
       "17   fungi       193187                             45720   \n",
       "18   fungi       179240                             45093   \n",
       "19   fungi       136650                             37159   \n",
       "20   fungi       141238                             35950   \n",
       "21   fungi       229073                             36666   \n",
       "22   fungi       125014                             54061   \n",
       "23   fungi       167818                             42373   \n",
       "24   fungi       249468                             42899   \n",
       "25   fungi       172601                             32847   \n",
       "26   fungi       198500                             29833   \n",
       "27   fungi       213143                             52944   \n",
       "28   fungi       223186                             59651   \n",
       "29   fungi       204777                             47318   \n",
       "30   fungi       205404                             49780   \n",
       "31   fungi       155158                             33941   \n",
       "32   fungi       206161                             42329   \n",
       "33   fungi       188353                             33438   \n",
       "34   fungi        96837                             33025   \n",
       "35   fungi       115477                             28063   \n",
       "36   fungi        91031                             26114   \n",
       "37   fungi       127633                             40257   \n",
       "38   fungi       104214                             32490   \n",
       "39   fungi       135720                             45958   \n",
       "40   fungi       109905                             38085   \n",
       "41   fungi       115282                             31150   \n",
       "42   fungi       122601                             35604   \n",
       "43   fungi       104844                             29540   \n",
       "\n",
       "    # reads after length filtering  # for use  \\\n",
       "0                           113337   113337.0   \n",
       "1                           133089   133089.0   \n",
       "2                            90015    90015.0   \n",
       "3                           123742   123742.0   \n",
       "4                            74232    74232.0   \n",
       "5                           117090   117090.0   \n",
       "6                            59406    59406.0   \n",
       "7                           131936   131936.0   \n",
       "8                           152736   152736.0   \n",
       "9                            72307    72307.0   \n",
       "10                          117801   117801.0   \n",
       "11                           93723    93723.0   \n",
       "12                           37315    37315.0   \n",
       "13                           33499    33499.0   \n",
       "14                            7805    16587.0   \n",
       "15                           36478    36478.0   \n",
       "16                           18944    18944.0   \n",
       "17                           42589    42589.0   \n",
       "18                           42369    42369.0   \n",
       "19                           34841    34841.0   \n",
       "20                           33873    33873.0   \n",
       "21                           34002    34002.0   \n",
       "22                           51340    51340.0   \n",
       "23                           39235    39235.0   \n",
       "24                           39988    39988.0   \n",
       "25                           30504    30504.0   \n",
       "26                           27941    27941.0   \n",
       "27                           49481    49481.0   \n",
       "28                           55589    55589.0   \n",
       "29                           44257    44257.0   \n",
       "30                           46171    46171.0   \n",
       "31                           31356    31356.0   \n",
       "32                           39836    39836.0   \n",
       "33                           31996    31996.0   \n",
       "34                           30260    30260.0   \n",
       "35                           25917    25917.0   \n",
       "36                           25597    25597.0   \n",
       "37                           37426    37426.0   \n",
       "38                           29765    29765.0   \n",
       "39                           42338    42338.0   \n",
       "40                           35120    35120.0   \n",
       "41                           28382    28382.0   \n",
       "42                           33244    33244.0   \n",
       "43                           26937    26937.0   \n",
       "\n",
       "                                    path to raw reads  \\\n",
       "0   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "1   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "2   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "3   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "4   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "5   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "6   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "7   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "8   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "9   analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "10  analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "11  analysis/Concatenated/20171103_FAH15473/barcod...   \n",
       "12  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "13  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "14  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "15  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "16  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "17  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "18  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "19  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "20  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "21  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "22  analysis/Concatenated/20171207_FAH18654/barcod...   \n",
       "23  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "24  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "25  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "26  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "27  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "28  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "29  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "30  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "31  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "32  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "33  analysis/Concatenated/20171212_FAH18688/barcod...   \n",
       "34  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "35  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "36  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "37  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "38  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "39  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "40  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "41  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "42  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "43  analysis/Concatenated/20180108_FAH18647/barcod...   \n",
       "\n",
       "                           path to homology filtering  \\\n",
       "0   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "1   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "2   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "3   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "4   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "5   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "6   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "7   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "8   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "9   analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "10  analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "11  analysis/Python_Processing/20171103_FAH15473/b...   \n",
       "12  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "13  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "14  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "15  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "16  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "17  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "18  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "19  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "20  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "21  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "22  analysis/Python_Processing/20171207_FAH18654/b...   \n",
       "23  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "24  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "25  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "26  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "27  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "28  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "29  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "30  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "31  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "32  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "33  analysis/Python_Processing/20171212_FAH18688/b...   \n",
       "34  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "35  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "36  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "37  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "38  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "39  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "40  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "41  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "42  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "43  analysis/Python_Processing/20180108_FAH18647/b...   \n",
       "\n",
       "                             path to length filtering  \\\n",
       "0   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "1   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "2   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "3   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "4   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "5   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "6   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "7   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "8   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "9   analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "10  analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "11  analysis/Length_Filtered/20171103_FAH15473/bar...   \n",
       "12  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "13  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "14  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "15  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "16  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "17  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "18  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "19  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "20  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "21  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "22  analysis/Length_Filtered/20171207_FAH18654/bar...   \n",
       "23  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "24  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "25  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "26  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "27  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "28  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "29  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "30  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "31  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "32  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "33  analysis/Length_Filtered/20171212_FAH18688/bar...   \n",
       "34  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "35  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "36  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "37  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "38  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "39  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "40  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "41  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "42  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "43  analysis/Length_Filtered/20180108_FAH18647/bar...   \n",
       "\n",
       "                                         path for use  \n",
       "0   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "1   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "2   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "3   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "4   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "5   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "6   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "7   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "8   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "9   analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "10  analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "11  analysis/Length_Filtered/20171103_FAH15473/bar...  \n",
       "12  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "13  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "14  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "15  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "16  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "17  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "18  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "19  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "20  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "21  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "22  analysis/Length_Filtered/20171207_FAH18654/bar...  \n",
       "23  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "24  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "25  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "26  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "27  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "28  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "29  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "30  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "31  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "32  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "33  analysis/Length_Filtered/20171212_FAH18688/bar...  \n",
       "34  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "35  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "36  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "37  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "38  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "39  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "40  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "41  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "42  analysis/Length_Filtered/20180108_FAH18647/bar...  \n",
       "43  analysis/Length_Filtered/20180108_FAH18647/bar...  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
