{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%writefile ../alignment_to_database.py\n",
    "\n",
    "# \"\"\"\n",
    "# Script aims to read in the reference_dataframe file, select a taxonomic\n",
    "# level and group, and read the path to the location of that data. It then\n",
    "# prepares data for machine learning by converting base pair coding to numerical\n",
    "# encoding, pads it out and then runs the algorithm\n",
    "# \"\"\"\n",
    "\n",
    "# import pandas as pd\n",
    "# from Bio import SeqIO\n",
    "# import numpy as np\n",
    "# import os\n",
    "# import random\n",
    "# import argparse\n",
    "# import math\n",
    "# import matplotlib.pyplot as plt\n",
    "# import pandas as pd\n",
    "\n",
    "# parser = argparse.ArgumentParser(description=\"\"\"\n",
    "# Script aims to read in the reference_dataframe file, and for each\n",
    "# sample, randomly subsample n_reads number of reads and save the\n",
    "# file, and save the keys used as as separate file. The script ends there.\n",
    "\n",
    "# Outside the script these files will be utilised in minimap2 to\n",
    "# map against a specified database, produce an output paf file\n",
    "# and then remove the file to conserve space. \n",
    "# \"\"\")\n",
    "# parser.add_argument(\"ref_df_fn\", help=\"File path to the reference dataframe\")\n",
    "# parser.add_argument(\"data_root\", help=\"Root folder for analysis/\")\n",
    "# parser.add_argument(\"--n_reads\", \"-c\", help=\"count of reads per class\")\n",
    "# parser.add_argument(\"--d_type\", \"-d\", help=\"type of database - CUSTOM or UNITE\")\n",
    "# group = parser.add_mutually_exclusive_group()\n",
    "# group.add_argument(\"--verbose\", \"-v\", \"--v\", action=\"store_true\")\n",
    "# group.add_argument(\"--quiet\", \"-q\", \"--q\", action=\"store_true\")\n",
    "# args = parser.parse_args()\n",
    "\n",
    "# # assign required arguments to variables\n",
    "# ref_df_fn = args.ref_df_fn\n",
    "# data_root = args.data_root\n",
    "\n",
    "# # assign a number of reads per class\n",
    "# n_reads = int(args.n_reads)\n",
    "\n",
    "# # test to make sure both required file paths are input\n",
    "# try:\n",
    "#     os.path.exists(ref_df_fn)\n",
    "# except:\n",
    "#     print('Cannot find %s' % ref_df_fn)\n",
    "# try:\n",
    "#     os.path.exists(data_root)\n",
    "# except:\n",
    "#     print('Cannot find %s' % data_root)\n",
    "    \n",
    "# if args.verbose:\n",
    "#     print('\\033[1;34m' + \"Reference dataframe is at \" + ref_df_fn + '\\033[0m')\n",
    "#     print('\\033[1;34m' + \"Root directory is at \" + data_root + '\\033[0m')\n",
    "#     print('\\033[1;34m' + \"Count of reads per sample is\", n_reads,'\\033[0m')\n",
    "    \n",
    "# # read in the reference dataframe from the argument path\n",
    "# ref_df = pd.read_csv(ref_df_fn, index_col=None)\n",
    "\n",
    "# # check whether the reference dataframe implies there are enough reads\n",
    "# # to continue given n_reads\n",
    "# try:\n",
    "#     if ref_df[ref_df[\"# for use\"] \\\n",
    "#               < n_reads].shape[0] > 0 :\n",
    "#         print(\"These species need more reads.\")\n",
    "#         print(ref_df[ref_df[\"# for use\"] \\\n",
    "#               < n_reads])\n",
    "#         exit()\n",
    "# except:\n",
    "#     print('Check %s to have the wanted column names' % ref_df_fn)\n",
    "   \n",
    "\n",
    "# for index, row in ref_df.iterrows():\n",
    "#     fasta_dict = SeqIO.to_dict(SeqIO.parse(row['path for use'], \"fasta\"))\n",
    "#     key_list = []\n",
    "#     for key in fasta_dict:\n",
    "#         key_list.append(key)\n",
    "#     print(len(key_list))\n",
    "#     keys_list = random.sample(key_list,k=n_reads)\n",
    "#     new_dict = {}\n",
    "#     for key in keys_list:\n",
    "#         new_dict[key] = fasta_dict[key]\n",
    "#     if args.d_type.upper() == 'CUSTOM':\n",
    "#         SeqIO.write(new_dict.values(), data_root+\"database_mapping/custom/%s_%s.fasta\" % (row['genus'],row['species']), \"fasta\")\n",
    "#         with open(data_root+'database_mapping/custom/%s_%s_keys.csv'% (row['genus'],row['species']), 'w+') as f:\n",
    "#             for key in keys_list:\n",
    "#                 f.write(\"%s\\n\"%(key))\n",
    "#     elif args.d_type.upper() == 'UNITE':\n",
    "#         SeqIO.write(new_dict.values(), \"analysis/database_mapping/unite/%s_%s.fasta\" % (row['genus'],row['species']), \"fasta\")\n",
    "#         with open(data_root+'database_mapping/unite/%s_%s_keys.csv'% (row['genus'],row['species']), 'w+') as f:\n",
    "#             for key in keys_list:\n",
    "#                 f.write(\"%s\\n\"%(key))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%writefile ../custom_minimap_result.py\n",
    "\n",
    "# import Bio\n",
    "# from Bio import SeqIO\n",
    "# from Bio.Seq import Seq\n",
    "# from Bio.SeqRecord import SeqRecord as SR\n",
    "# from Bio.Blast import NCBIXML\n",
    "# from Bio import pairwise2\n",
    "# from Bio.pairwise2 import format_alignment\n",
    "# from Bio import AlignIO\n",
    "# from Bio.Align import AlignInfo\n",
    "# import numpy as np\n",
    "# import pandas as pd\n",
    "# from pandas import DataFrame as df\n",
    "# # import seaborn as sns\n",
    "# import csv\n",
    "# import matplotlib\n",
    "# import matplotlib.pyplot as plt\n",
    "# import subprocess\n",
    "# import os\n",
    "# from shutil import copy\n",
    "# import random\n",
    "# import warnings\n",
    "# from Bio import pairwise2\n",
    "# from Bio.pairwise2 import format_alignment\n",
    "# # from ipysankeywidget import SankeyWidget\n",
    "# warnings.filterwarnings(\"ignore\")\n",
    "# import argparse\n",
    "\n",
    "\n",
    "# parser = argparse.ArgumentParser(description=\"\"\"\n",
    "# This program extracts a specified number of reads from a fasta\n",
    "# file and saves them to a new fasta file\n",
    "# \"\"\")\n",
    "# group = parser.add_mutually_exclusive_group()\n",
    "# group.add_argument(\"--verbose\", \"-v\", \"--v\", action=\"store_true\")\n",
    "# group.add_argument(\"--quiet\", \"-q\", \"--q\", action=\"store_true\")\n",
    "# parser.add_argument(\"input_file\", help=\"The input file for extraction\")\n",
    "# args = parser.parse_args()\n",
    "\n",
    "# if args.verbose:\n",
    "#     print(\"Input file is \" + args.input_file + \"\\n\")\n",
    "\n",
    "# f=open(args.input_file,\"r\")\n",
    "# if f.mode == \"r\":\n",
    "#     contents=f.read()\n",
    "\n",
    "# tmp=contents.replace(\"\\t\",\",\").split('\\n')\n",
    "# tmp_dict = {}\n",
    "# for line in tmp[:-1]:\n",
    "#     tmp_dict[line.split(\",\")[0]] = str(line.split(\",\")[5])\n",
    "\n",
    "# count_dict = {}\n",
    "# correct = 0\n",
    "# incorrect = 0\n",
    "# for item in tmp_dict:\n",
    "#     if tmp_dict[item] not in count_dict:\n",
    "#         count_dict[tmp_dict[item]] = 1\n",
    "#     else:\n",
    "#         count_dict[tmp_dict[item]] = count_dict[tmp_dict[item]] + 1\n",
    "        \n",
    "# tmp = pd.DataFrame.from_dict(count_dict,orient='index',columns=[\"Count\"])\n",
    "# tmp.index.names = [args.input_file[33:-4]]\n",
    "# tmp = tmp.sort_values(by=\"Count\",ascending=False)\n",
    "\n",
    "# if args.verbose:\n",
    "#     print(tmp)\n",
    "\n",
    "# tmp.to_csv(args.input_file[:-4]+'_match_distribution.csv',sep=',')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Starting points from Tavish  \n",
    "\n",
    "reference_dataframe at '/media/MassStorage/tmp/TE/honours/analysis/Stats/reference_dataframe.csv' \n",
    "\n",
    "custom_database at '/media/MassStorage/tmp/TE/honours/database/custom_database_labelled.fasta'\n",
    "\n",
    "taxonomy_file at '/media/MassStorage/tmp/TE/honours/analysis/Stats/taxonomy_file.csv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### workflow\n",
    "\n",
    "* Two databases\n",
    "* subsample 15000 reads per each mock community species. Save those out.\n",
    "* map reads against both databases with minimap safe out data in paf format.\n",
    "* get best hit per species (see what this means while looking at the data).\n",
    "* add the full taxonomy to each best match using the taxonomy file.\n",
    "* summarize data at different taxonomic ranks for each species.\n",
    "* pull this all together somehow (summary across all the samples? focus on species of interest e.g. deleted from analyis?)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "import random\n",
    "import subprocess\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "reference_dataframe_fn = os.path.abspath('/media/MassStorage/tmp/TE/honours/analysis/Stats/reference_dataframe.csv')\n",
    "max_custom_database_fn = os.path.abspath('/media/MassStorage/tmp/TE/honours/database/custom_database_labelled.fasta')\n",
    "taxonomy_file_fn = os.path.abspath('/media/MassStorage/tmp/TE/honours/analysis/Stats/taxonomy_file.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "INPUT_BASEDIR = os.path.abspath('/media/MassStorage/tmp/TE/honours')\n",
    "\n",
    "OUT_DIR = os.path.abspath('../../analysis/Mapping')\n",
    "if not os.path.exists(OUT_DIR):\n",
    "    os.mkdir(OUT_DIR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "### list of species in the max database\n",
    "max_species = ['Puccinia_striiformis-tritici',\n",
    "             'Zymoseptoria_tritici',\n",
    "             'Pyrenophora_tritici-repentis',\n",
    "             'Fusarium_oxysporum',\n",
    "             'Tuber_brumale',\n",
    "             'Cortinarius_globuliformis',\n",
    "             'Aspergillus_niger',\n",
    "             'Clavispora_lusitaniae',\n",
    "             'Kluyveromyces_unidentified',\n",
    "             'Penicillium_chrysogenum',\n",
    "             'Rhodotorula_mucilaginosa',\n",
    "             'Scedosporium_boydii',\n",
    "             'Blastobotrys_proliferans',\n",
    "             'Debaryomyces_unidentified',\n",
    "             'Galactomyces_geotrichum',\n",
    "             'Kodamaea_ohmeri',\n",
    "             'Meyerozyma_guillermondii',\n",
    "             'Wickerhamomyces_anomalus',\n",
    "             'Yamadazyma_mexicana',\n",
    "             'Yamadazyma_scolyti',\n",
    "             'Yarrowia_lipolytica',\n",
    "             'Zygoascus_hellenicus',\n",
    "             'Aspergillus_flavus',\n",
    "             'Cryptococcus_zero',\n",
    "             'Aspergillus_unidentified',\n",
    "             'Diaporthe_CCL067',\n",
    "             'Diaporthe_unidentified',\n",
    "             'Oculimacula_yallundae-CCL031',\n",
    "             'Oculimacula_yallundae-CCL029',\n",
    "             'Dothiorella_vidmadera',\n",
    "             'Quambalaria_cyanescens',\n",
    "             'Entoleuca_unidentified',\n",
    "             'Asteroma_CCL060',\n",
    "             'Asteroma_CCL068',\n",
    "             'Saccharomyces_cerevisiae',\n",
    "             'Cladophialophora_unidentified',\n",
    "             'Candida_albicans',\n",
    "             'Candida_metapsilosis',\n",
    "             'Candida_orthopsilosis',\n",
    "             'Candida_parapsilosis',\n",
    "             'Candida_unidentified',\n",
    "             'Kluyveromyces_marxianus',\n",
    "             'Pichia_kudriavzevii',\n",
    "             'Pichia_membranifaciens']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "###Removed from second test database\n",
    "species_delete = ['Candida_orthopsilosis',\n",
    "                 'Candida_metapsilosis',\n",
    "                 'Aspergillus_niger',\n",
    "                 'Yamadazyma_mexicana']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "###species to be searched against both databases\n",
    "mock_community = ['Puccinia_striiformis-tritici',\n",
    "             'Zymoseptoria_tritici',\n",
    "             'Pyrenophora_tritici-repentis',\n",
    "             'Fusarium_oxysporum',\n",
    "             'Tuber_brumale',\n",
    "             'Cortinarius_globuliformis',\n",
    "             'Aspergillus_niger',\n",
    "             'Clavispora_lusitaniae',\n",
    "             'Kluyveromyces_unidentified',\n",
    "             'Penicillium_chrysogenum',\n",
    "             'Rhodotorula_mucilaginosa',\n",
    "             'Scedosporium_boydii',\n",
    "             'Blastobotrys_proliferans',\n",
    "             'Debaryomyces_unidentified',\n",
    "             'Galactomyces_geotrichum',\n",
    "             'Kodamaea_ohmeri',\n",
    "             'Meyerozyma_guillermondii',\n",
    "             'Wickerhamomyces_anomalus',\n",
    "             'Yamadazyma_mexicana',\n",
    "             'Yamadazyma_scolyti',\n",
    "             'Yarrowia_lipolytica',\n",
    "             'Zygoascus_hellenicus',\n",
    "             'Aspergillus_flavus',\n",
    "             'Cryptococcus_zero',\n",
    "             'Aspergillus_unidentified',\n",
    "             'Diaporthe_CCL067',\n",
    "             'Diaporthe_unidentified',\n",
    "             'Oculimacula_yallundae-CCL031',\n",
    "             'Oculimacula_yallundae-CCL029',\n",
    "             'Dothiorella_vidmadera',\n",
    "             'Quambalaria_cyanescens',\n",
    "             'Entoleuca_unidentified',\n",
    "             'Asteroma_CCL060',\n",
    "             'Asteroma_CCL068',\n",
    "             'Saccharomyces_cerevisiae',\n",
    "             'Cladophialophora_unidentified',\n",
    "             'Candida_albicans',\n",
    "             'Candida_metapsilosis',\n",
    "             'Candida_orthopsilosis',\n",
    "             'Candida_parapsilosis',\n",
    "             'Candida_unidentified',\n",
    "             'Kluyveromyces_marxianus',\n",
    "             'Pichia_kudriavzevii',\n",
    "             'Pichia_membranifaciens']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df = pd.read_csv(reference_dataframe_fn)\n",
    "ref_df['name_species'] = ref_df['genus'] +\"_\"+ ref_df['species']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['puccinia_striiformis-tritici',\n",
       " 'zymoseptoria_tritici',\n",
       " 'pyrenophora_tritici-repentis',\n",
       " 'fusarium_oxysporum',\n",
       " 'tuber_brumale',\n",
       " 'cortinarius_globuliformis',\n",
       " 'aspergillus_niger',\n",
       " 'clavispora_lusitaniae',\n",
       " 'kluyveromyces_unidentified',\n",
       " 'penicillium_chrysogenum',\n",
       " 'rhodotorula_mucilaginosa',\n",
       " 'scedosporium_boydii',\n",
       " 'blastobotrys_proliferans',\n",
       " 'debaryomyces_unidentified',\n",
       " 'galactomyces_geotrichum',\n",
       " 'kodamaea_ohmeri',\n",
       " 'meyerozyma_guillermondii',\n",
       " 'wickerhamomyces_anomalus',\n",
       " 'yamadazyma_mexicana',\n",
       " 'yamadazyma_scolyti',\n",
       " 'yarrowia_lipolytica',\n",
       " 'zygoascus_hellenicus',\n",
       " 'aspergillus_flavus',\n",
       " 'cryptococcus_zero',\n",
       " 'aspergillus_unidentified',\n",
       " 'diaporthe_ccl067',\n",
       " 'diaporthe_unidentified',\n",
       " 'oculimacula_yallundae-ccl031',\n",
       " 'oculimacula_yallundae-ccl029',\n",
       " 'dothiorella_vidmadera',\n",
       " 'quambalaria_cyanescens',\n",
       " 'entoleuca_unidentified',\n",
       " 'asteroma_ccl060',\n",
       " 'asteroma_ccl068',\n",
       " 'saccharomyces_cerevisiae',\n",
       " 'cladophialophora_unidentified',\n",
       " 'candida_albicans',\n",
       " 'candida_metapsilosis',\n",
       " 'candida_orthopsilosis',\n",
       " 'candida_parapsilosis',\n",
       " 'candida_unidentified',\n",
       " 'kluyveromyces_marxianus',\n",
       " 'pichia_kudriavzevii',\n",
       " 'pichia_membranifaciens']"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_df.name_species.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_db_fn = os.path.join(OUT_DIR, 'gsref.db.fasta')\n",
    "\n",
    "new_db_list = []\n",
    "old_db_list = []\n",
    "for seq in SeqIO.parse(max_custom_database_fn, 'fasta'):\n",
    "    old_db_list.append(seq.id)\n",
    "    if seq.id.lower() in ref_df.name_species.tolist():\n",
    "        #print(seq.id)\n",
    "        seq.id = seq.name = seq.description = seq.id.lower()\n",
    "        new_db_list.append(seq)\n",
    "    else:\n",
    "        print(seq.id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "if len(new_db_list) == len(old_db_list):\n",
    "    SeqIO.write(new_db_list, new_db_fn, 'fasta')\n",
    "else:\n",
    "    print(\"please check!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "sub_db_fn = os.path.join(OUT_DIR, 'gsref.subdb.fasta')\n",
    "sub_db_list = []\n",
    "for seq in new_db_list:\n",
    "    if seq.id not in [x.lower() for x in species_delete]:\n",
    "        sub_db_list.append(seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "if len(sub_db_list) + len(species_delete) == len(new_db_list):\n",
    "    SeqIO.write(sub_db_list, sub_db_fn, 'fasta' )\n",
    "else:\n",
    "    print(\"please check!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['puccinia_striiformis-tritici',\n",
       " 'zymoseptoria_tritici',\n",
       " 'pyrenophora_tritici-repentis',\n",
       " 'fusarium_oxysporum',\n",
       " 'tuber_brumale',\n",
       " 'cortinarius_globuliformis',\n",
       " 'clavispora_lusitaniae',\n",
       " 'kluyveromyces_unidentified',\n",
       " 'penicillium_chrysogenum',\n",
       " 'rhodotorula_mucilaginosa',\n",
       " 'scedosporium_boydii',\n",
       " 'blastobotrys_proliferans',\n",
       " 'debaryomyces_unidentified',\n",
       " 'galactomyces_geotrichum',\n",
       " 'kodamaea_ohmeri',\n",
       " 'meyerozyma_guillermondii',\n",
       " 'wickerhamomyces_anomalus',\n",
       " 'yamadazyma_scolyti',\n",
       " 'yarrowia_lipolytica',\n",
       " 'zygoascus_hellenicus',\n",
       " 'aspergillus_flavus',\n",
       " 'cryptococcus_zero',\n",
       " 'aspergillus_unidentified',\n",
       " 'diaporthe_ccl067',\n",
       " 'diaporthe_unidentified',\n",
       " 'oculimacula_yallundae-ccl031',\n",
       " 'oculimacula_yallundae-ccl029',\n",
       " 'dothiorella_vidmadera',\n",
       " 'quambalaria_cyanescens',\n",
       " 'entoleuca_unidentified',\n",
       " 'asteroma_ccl060',\n",
       " 'asteroma_ccl068',\n",
       " 'saccharomyces_cerevisiae',\n",
       " 'cladophialophora_unidentified',\n",
       " 'candida_albicans',\n",
       " 'candida_parapsilosis',\n",
       " 'candida_unidentified',\n",
       " 'kluyveromyces_marxianus',\n",
       " 'pichia_kudriavzevii',\n",
       " 'pichia_membranifaciens']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x.id for x in sub_db_list]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "mock_community = [x.lower() for x in mock_community]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['puccinia_striiformis-tritici',\n",
       " 'zymoseptoria_tritici',\n",
       " 'pyrenophora_tritici-repentis',\n",
       " 'fusarium_oxysporum',\n",
       " 'tuber_brumale',\n",
       " 'cortinarius_globuliformis',\n",
       " 'aspergillus_niger',\n",
       " 'clavispora_lusitaniae',\n",
       " 'kluyveromyces_unidentified',\n",
       " 'penicillium_chrysogenum',\n",
       " 'rhodotorula_mucilaginosa',\n",
       " 'scedosporium_boydii',\n",
       " 'blastobotrys_proliferans',\n",
       " 'debaryomyces_unidentified',\n",
       " 'galactomyces_geotrichum',\n",
       " 'kodamaea_ohmeri',\n",
       " 'meyerozyma_guillermondii',\n",
       " 'wickerhamomyces_anomalus',\n",
       " 'yamadazyma_mexicana',\n",
       " 'yamadazyma_scolyti',\n",
       " 'yarrowia_lipolytica',\n",
       " 'zygoascus_hellenicus',\n",
       " 'aspergillus_flavus',\n",
       " 'cryptococcus_zero',\n",
       " 'aspergillus_unidentified',\n",
       " 'diaporthe_ccl067',\n",
       " 'diaporthe_unidentified',\n",
       " 'oculimacula_yallundae-ccl031',\n",
       " 'oculimacula_yallundae-ccl029',\n",
       " 'dothiorella_vidmadera',\n",
       " 'quambalaria_cyanescens',\n",
       " 'entoleuca_unidentified',\n",
       " 'asteroma_ccl060',\n",
       " 'asteroma_ccl068',\n",
       " 'saccharomyces_cerevisiae',\n",
       " 'cladophialophora_unidentified',\n",
       " 'candida_albicans',\n",
       " 'candida_metapsilosis',\n",
       " 'candida_orthopsilosis',\n",
       " 'candida_parapsilosis',\n",
       " 'candida_unidentified',\n",
       " 'kluyveromyces_marxianus',\n",
       " 'pichia_kudriavzevii',\n",
       " 'pichia_membranifaciens']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mock_community"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "def subsamplereads(in_fn, out_fn, n_reads):\n",
    "    command = F'reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}'\n",
    "    out = subprocess.getstatusoutput(command)\n",
    "    if out[0] == 0:\n",
    "        print(F\":)Completed {command}\\n\")\n",
    "    else:\n",
    "        print(F\":(check one {command}!!\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_reads = 15000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "MC_READ_DIR = os.path.join(OUT_DIR, 'MC_READS')\n",
    "if not os.path.exists(MC_READ_DIR):\n",
    "    os.mkdir(MC_READ_DIR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'species', 'genus', 'family', 'order', 'class', 'phylum',\n",
       "       'kingdom', '# raw reads', '# reads after homology filtering',\n",
       "       '# reads after length filtering', '# for use', 'path to raw reads',\n",
       "       'path to homology filtering', 'path to length filtering',\n",
       "       'path for use', 'name_species'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'puccinia_striiformis-tritici': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode01/length_restricted_for_use.fasta',\n",
       " 'zymoseptoria_tritici': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode02/length_restricted_for_use.fasta',\n",
       " 'pyrenophora_tritici-repentis': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode03/length_restricted_for_use.fasta',\n",
       " 'fusarium_oxysporum': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode04/length_restricted_for_use.fasta',\n",
       " 'tuber_brumale': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode05/length_restricted_for_use.fasta',\n",
       " 'cortinarius_globuliformis': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode06/length_restricted_for_use.fasta',\n",
       " 'aspergillus_niger': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode07/length_restricted_for_use.fasta',\n",
       " 'clavispora_lusitaniae': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode08/length_restricted_for_use.fasta',\n",
       " 'kluyveromyces_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode09/length_restricted_for_use.fasta',\n",
       " 'penicillium_chrysogenum': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode10/length_restricted_for_use.fasta',\n",
       " 'rhodotorula_mucilaginosa': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode11/length_restricted_for_use.fasta',\n",
       " 'scedosporium_boydii': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode12/length_restricted_for_use.fasta',\n",
       " 'blastobotrys_proliferans': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode01/length_restricted_for_use.fasta',\n",
       " 'debaryomyces_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode02/length_restricted_for_use.fasta',\n",
       " 'galactomyces_geotrichum': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode03/length_restricted_for_use.fasta',\n",
       " 'kodamaea_ohmeri': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode04/length_restricted_for_use.fasta',\n",
       " 'meyerozyma_guillermondii': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode05/length_restricted_for_use.fasta',\n",
       " 'wickerhamomyces_anomalus': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode06/length_restricted_for_use.fasta',\n",
       " 'yamadazyma_mexicana': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode07/length_restricted_for_use.fasta',\n",
       " 'yamadazyma_scolyti': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode08/length_restricted_for_use.fasta',\n",
       " 'yarrowia_lipolytica': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode09/length_restricted_for_use.fasta',\n",
       " 'zygoascus_hellenicus': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode11/length_restricted_for_use.fasta',\n",
       " 'aspergillus_flavus': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode12/length_restricted_for_use.fasta',\n",
       " 'cryptococcus_zero': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode01/length_restricted_for_use.fasta',\n",
       " 'aspergillus_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode02/length_restricted_for_use.fasta',\n",
       " 'diaporthe_ccl067': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode03/length_restricted_for_use.fasta',\n",
       " 'diaporthe_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode04/length_restricted_for_use.fasta',\n",
       " 'oculimacula_yallundae-ccl031': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode05/length_restricted_for_use.fasta',\n",
       " 'oculimacula_yallundae-ccl029': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode06/length_restricted_for_use.fasta',\n",
       " 'dothiorella_vidmadera': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode07/length_restricted_for_use.fasta',\n",
       " 'quambalaria_cyanescens': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode08/length_restricted_for_use.fasta',\n",
       " 'entoleuca_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode09/length_restricted_for_use.fasta',\n",
       " 'asteroma_ccl060': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode11/length_restricted_for_use.fasta',\n",
       " 'asteroma_ccl068': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171212_FAH18688/barcode12/length_restricted_for_use.fasta',\n",
       " 'saccharomyces_cerevisiae': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode01/length_restricted_for_use.fasta',\n",
       " 'cladophialophora_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode02/length_restricted_for_use.fasta',\n",
       " 'candida_albicans': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode03/length_restricted_for_use.fasta',\n",
       " 'candida_metapsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode04/length_restricted_for_use.fasta',\n",
       " 'candida_orthopsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode05/length_restricted_for_use.fasta',\n",
       " 'candida_parapsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode06/length_restricted_for_use.fasta',\n",
       " 'candida_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode09/length_restricted_for_use.fasta',\n",
       " 'kluyveromyces_marxianus': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode10/length_restricted_for_use.fasta',\n",
       " 'pichia_kudriavzevii': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode11/length_restricted_for_use.fasta',\n",
       " 'pichia_membranifaciens': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode12/length_restricted_for_use.fasta'}"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fn_subsampling = {}\n",
    "for x in mock_community:\n",
    "    fn_subsampling[x] = (ref_df[(ref_df['species'] == x.split('_')[1]) & (ref_df['genus'] == x.split('_')[0])]['path for use'].tolist()[0])\n",
    "    fn_subsampling[x] = os.path.join(INPUT_BASEDIR, fn_subsampling[x])\n",
    "fn_subsampling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/puccinia_striiformis-tritici.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zymoseptoria_tritici.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pyrenophora_tritici-repentis.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/fusarium_oxysporum.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/tuber_brumale.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cortinarius_globuliformis.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_niger.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/clavispora_lusitaniae.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_unidentified.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/penicillium_chrysogenum.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/rhodotorula_mucilaginosa.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/scedosporium_boydii.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/blastobotrys_proliferans.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/debaryomyces_unidentified.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/galactomyces_geotrichum.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kodamaea_ohmeri.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/meyerozyma_guillermondii.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/wickerhamomyces_anomalus.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_mexicana.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_scolyti.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yarrowia_lipolytica.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zygoascus_hellenicus.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_flavus.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cryptococcus_zero.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_unidentified.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_ccl067.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_unidentified.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl031.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl029.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/dothiorella_vidmadera.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/quambalaria_cyanescens.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/entoleuca_unidentified.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl060.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl068.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/saccharomyces_cerevisiae.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cladophialophora_unidentified.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_albicans.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_metapsilosis.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_orthopsilosis.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_parapsilosis.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_unidentified.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_marxianus.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_kudriavzevii.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n",
      "/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_membranifaciens.15000.fasta\n",
      ":(check one reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}!!\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'puccinia_striiformis-tritici': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/puccinia_striiformis-tritici.15000.fasta',\n",
       " 'zymoseptoria_tritici': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zymoseptoria_tritici.15000.fasta',\n",
       " 'pyrenophora_tritici-repentis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pyrenophora_tritici-repentis.15000.fasta',\n",
       " 'fusarium_oxysporum': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/fusarium_oxysporum.15000.fasta',\n",
       " 'tuber_brumale': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/tuber_brumale.15000.fasta',\n",
       " 'cortinarius_globuliformis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cortinarius_globuliformis.15000.fasta',\n",
       " 'aspergillus_niger': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_niger.15000.fasta',\n",
       " 'clavispora_lusitaniae': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/clavispora_lusitaniae.15000.fasta',\n",
       " 'kluyveromyces_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_unidentified.15000.fasta',\n",
       " 'penicillium_chrysogenum': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/penicillium_chrysogenum.15000.fasta',\n",
       " 'rhodotorula_mucilaginosa': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/rhodotorula_mucilaginosa.15000.fasta',\n",
       " 'scedosporium_boydii': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/scedosporium_boydii.15000.fasta',\n",
       " 'blastobotrys_proliferans': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/blastobotrys_proliferans.15000.fasta',\n",
       " 'debaryomyces_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/debaryomyces_unidentified.15000.fasta',\n",
       " 'galactomyces_geotrichum': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/galactomyces_geotrichum.15000.fasta',\n",
       " 'kodamaea_ohmeri': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kodamaea_ohmeri.15000.fasta',\n",
       " 'meyerozyma_guillermondii': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/meyerozyma_guillermondii.15000.fasta',\n",
       " 'wickerhamomyces_anomalus': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/wickerhamomyces_anomalus.15000.fasta',\n",
       " 'yamadazyma_mexicana': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_mexicana.15000.fasta',\n",
       " 'yamadazyma_scolyti': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_scolyti.15000.fasta',\n",
       " 'yarrowia_lipolytica': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yarrowia_lipolytica.15000.fasta',\n",
       " 'zygoascus_hellenicus': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zygoascus_hellenicus.15000.fasta',\n",
       " 'aspergillus_flavus': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_flavus.15000.fasta',\n",
       " 'cryptococcus_zero': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cryptococcus_zero.15000.fasta',\n",
       " 'aspergillus_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_unidentified.15000.fasta',\n",
       " 'diaporthe_ccl067': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_ccl067.15000.fasta',\n",
       " 'diaporthe_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_unidentified.15000.fasta',\n",
       " 'oculimacula_yallundae-ccl031': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl031.15000.fasta',\n",
       " 'oculimacula_yallundae-ccl029': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl029.15000.fasta',\n",
       " 'dothiorella_vidmadera': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/dothiorella_vidmadera.15000.fasta',\n",
       " 'quambalaria_cyanescens': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/quambalaria_cyanescens.15000.fasta',\n",
       " 'entoleuca_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/entoleuca_unidentified.15000.fasta',\n",
       " 'asteroma_ccl060': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl060.15000.fasta',\n",
       " 'asteroma_ccl068': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl068.15000.fasta',\n",
       " 'saccharomyces_cerevisiae': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/saccharomyces_cerevisiae.15000.fasta',\n",
       " 'cladophialophora_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cladophialophora_unidentified.15000.fasta',\n",
       " 'candida_albicans': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_albicans.15000.fasta',\n",
       " 'candida_metapsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_metapsilosis.15000.fasta',\n",
       " 'candida_orthopsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_orthopsilosis.15000.fasta',\n",
       " 'candida_parapsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_parapsilosis.15000.fasta',\n",
       " 'candida_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_unidentified.15000.fasta',\n",
       " 'kluyveromyces_marxianus': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_marxianus.15000.fasta',\n",
       " 'pichia_kudriavzevii': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_kudriavzevii.15000.fasta',\n",
       " 'pichia_membranifaciens': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_membranifaciens.15000.fasta'}"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_reads_fn = {}\n",
    "for key, value in fn_subsampling.items():\n",
    "    species = key\n",
    "    in_fn = value\n",
    "    out_fn = os.path.join(MC_READ_DIR, F'{species}.{n_reads}.fasta')\n",
    "    print(out_fn)\n",
    "    subsamplereads(in_fn, out_fn, n_reads)\n",
    "    sub_reads_fn[species] = out_fn\n",
    "sub_reads_fn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Map with minimap against both databases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "def minimapmapping(fasta_fn, ref_fn, out_fn):\n",
    "    command = F\"minimap2 -x map-ont -t 6 {ref_fn} {fasta_fn} -o {out_fn}\"\n",
    "    out = subprocess.getstatusoutput(command)\n",
    "    print(out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb',\n",
       " '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db'}"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dbases_fn = {}\n",
    "for x in [sub_db_fn, new_db_fn]:\n",
    "    dbases_fn[x] = os.path.join(OUT_DIR, os.path.basename(x).replace('.fasta', '').replace('.','_'))\n",
    "    if not os.path.exists(dbases_fn[x]):\n",
    "        os.mkdir(dbases_fn[x])\n",
    "dbases_fn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.011*1.18] collected minimizers\\n[M::mm_idx_gen::0.013*1.97] sorted minimizers\\n[M::main::0.013*1.97] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.92] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.015*1.88] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/puccinia_striiformis-tritici.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.puccinia_striiformis-tritici.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/puccinia_striiformis-tritici.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.18] collected minimizers\\n[M::mm_idx_gen::0.012*2.12] sorted minimizers\\n[M::main::0.012*2.11] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.03] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*1.97] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zymoseptoria_tritici.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.zymoseptoria_tritici.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zymoseptoria_tritici.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.21] collected minimizers\\n[M::mm_idx_gen::0.011*2.37] sorted minimizers\\n[M::main::0.011*2.37] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.26] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.19] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pyrenophora_tritici-repentis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.pyrenophora_tritici-repentis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pyrenophora_tritici-repentis.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.006*1.21] collected minimizers\\n[M::mm_idx_gen::0.009*2.67] sorted minimizers\\n[M::main::0.009*2.66] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.52] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.011*2.43] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/fusarium_oxysporum.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.fusarium_oxysporum.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/fusarium_oxysporum.15000.fasta\\n[M::main] Real time: 0.013 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.19] collected minimizers\\n[M::mm_idx_gen::0.011*2.29] sorted minimizers\\n[M::main::0.011*2.29] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.19] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.13] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/tuber_brumale.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.tuber_brumale.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/tuber_brumale.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.24] collected minimizers\\n[M::mm_idx_gen::0.010*2.41] sorted minimizers\\n[M::main::0.010*2.41] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.26] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.16] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cortinarius_globuliformis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.cortinarius_globuliformis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cortinarius_globuliformis.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.20] collected minimizers\\n[M::mm_idx_gen::0.012*2.22] sorted minimizers\\n[M::main::0.012*2.22] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.13] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*2.06] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_niger.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.aspergillus_niger.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_niger.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.17] collected minimizers\\n[M::mm_idx_gen::0.013*2.06] sorted minimizers\\n[M::main::0.013*2.06] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.99] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.015*1.95] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/clavispora_lusitaniae.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.clavispora_lusitaniae.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/clavispora_lusitaniae.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.18] collected minimizers\\n[M::mm_idx_gen::0.012*2.25] sorted minimizers\\n[M::main::0.012*2.25] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.14] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*2.06] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.kluyveromyces_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_unidentified.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.032 sec; Peak RSS: 0.005 GB\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.010*1.12] collected minimizers\\n[M::mm_idx_gen::0.013*1.93] sorted minimizers\\n[M::main::0.013*1.93] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.88] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*1.84] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/penicillium_chrysogenum.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.penicillium_chrysogenum.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/penicillium_chrysogenum.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.006*1.26] collected minimizers\\n[M::mm_idx_gen::0.009*2.54] sorted minimizers\\n[M::main::0.009*2.54] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.37] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.011*2.27] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/rhodotorula_mucilaginosa.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.rhodotorula_mucilaginosa.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/rhodotorula_mucilaginosa.15000.fasta\\n[M::main] Real time: 0.013 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.19] collected minimizers\\n[M::mm_idx_gen::0.011*2.24] sorted minimizers\\n[M::main::0.011*2.24] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.14] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.07] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/scedosporium_boydii.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.scedosporium_boydii.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/scedosporium_boydii.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.24] collected minimizers\\n[M::mm_idx_gen::0.010*2.43] sorted minimizers\\n[M::main::0.010*2.43] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.31] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.23] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/blastobotrys_proliferans.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.blastobotrys_proliferans.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/blastobotrys_proliferans.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.005*1.32] collected minimizers\\n[M::mm_idx_gen::0.008*2.65] sorted minimizers\\n[M::main::0.008*2.65] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.009*2.44] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.010*2.30] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/debaryomyces_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.debaryomyces_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/debaryomyces_unidentified.15000.fasta\\n[M::main] Real time: 0.012 sec; CPU: 0.025 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.16] collected minimizers\\n[M::mm_idx_gen::0.013*1.98] sorted minimizers\\n[M::main::0.013*1.98] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.90] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.015*1.85] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/galactomyces_geotrichum.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.galactomyces_geotrichum.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/galactomyces_geotrichum.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.006*1.26] collected minimizers\\n[M::mm_idx_gen::0.009*2.56] sorted minimizers\\n[M::main::0.009*2.55] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.40] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.011*2.29] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kodamaea_ohmeri.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.kodamaea_ohmeri.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kodamaea_ohmeri.15000.fasta\\n[M::main] Real time: 0.013 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.23] collected minimizers\\n[M::mm_idx_gen::0.011*2.27] sorted minimizers\\n[M::main::0.011*2.27] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.17] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.10] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/meyerozyma_guillermondii.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.meyerozyma_guillermondii.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/meyerozyma_guillermondii.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.22] collected minimizers\\n[M::mm_idx_gen::0.010*2.30] sorted minimizers\\n[M::main::0.010*2.30] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.17] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.08] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/wickerhamomyces_anomalus.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.wickerhamomyces_anomalus.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/wickerhamomyces_anomalus.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.010*1.12] collected minimizers\\n[M::mm_idx_gen::0.012*2.07] sorted minimizers\\n[M::main::0.012*2.07] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.00] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*1.95] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_mexicana.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.yamadazyma_mexicana.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_mexicana.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.15] collected minimizers\\n[M::mm_idx_gen::0.013*2.06] sorted minimizers\\n[M::main::0.013*2.06] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.99] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.015*1.94] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_scolyti.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.yamadazyma_scolyti.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_scolyti.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.27] collected minimizers\\n[M::mm_idx_gen::0.010*2.44] sorted minimizers\\n[M::main::0.010*2.43] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.29] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.19] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yarrowia_lipolytica.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.yarrowia_lipolytica.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yarrowia_lipolytica.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.22] collected minimizers\\n[M::mm_idx_gen::0.011*2.32] sorted minimizers\\n[M::main::0.011*2.31] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.21] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.15] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zygoascus_hellenicus.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.zygoascus_hellenicus.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zygoascus_hellenicus.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.16] collected minimizers\\n[M::mm_idx_gen::0.013*2.09] sorted minimizers\\n[M::main::0.013*2.09] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.014*2.01] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.015*1.95] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_flavus.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.aspergillus_flavus.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_flavus.15000.fasta\\n[M::main] Real time: 0.018 sec; CPU: 0.032 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.17] collected minimizers\\n[M::mm_idx_gen::0.012*2.11] sorted minimizers\\n[M::main::0.012*2.11] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.06] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*2.02] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cryptococcus_zero.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.cryptococcus_zero.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cryptococcus_zero.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.005*1.29] collected minimizers\\n[M::mm_idx_gen::0.007*2.65] sorted minimizers\\n[M::main::0.007*2.65] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.008*2.48] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.009*2.36] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.aspergillus_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_unidentified.15000.fasta\\n[M::main] Real time: 0.011 sec; CPU: 0.023 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.15] collected minimizers\\n[M::mm_idx_gen::0.013*2.00] sorted minimizers\\n[M::main::0.013*2.00] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.92] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.015*1.86] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_ccl067.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.diaporthe_ccl067.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_ccl067.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.21] collected minimizers\\n[M::mm_idx_gen::0.011*2.29] sorted minimizers\\n[M::main::0.011*2.29] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.20] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.14] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.diaporthe_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_unidentified.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.17] collected minimizers\\n[M::mm_idx_gen::0.013*2.10] sorted minimizers\\n[M::main::0.013*2.10] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.014*2.02] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.015*1.97] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl031.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.oculimacula_yallundae-ccl031.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl031.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.009*1.18] collected minimizers\\n[M::mm_idx_gen::0.012*2.08] sorted minimizers\\n[M::main::0.012*2.08] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*1.99] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*1.93] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl029.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.oculimacula_yallundae-ccl029.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl029.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.23] collected minimizers\\n[M::mm_idx_gen::0.010*2.35] sorted minimizers\\n[M::main::0.010*2.35] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.21] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.12] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/dothiorella_vidmadera.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.dothiorella_vidmadera.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/dothiorella_vidmadera.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.14] collected minimizers\\n[M::mm_idx_gen::0.011*2.23] sorted minimizers\\n[M::main::0.011*2.23] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.17] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.12] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/quambalaria_cyanescens.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.quambalaria_cyanescens.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/quambalaria_cyanescens.15000.fasta\\n[M::main] Real time: 0.013 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.19] collected minimizers\\n[M::mm_idx_gen::0.012*2.18] sorted minimizers\\n[M::main::0.012*2.18] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.07] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*2.01] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/entoleuca_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.entoleuca_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/entoleuca_unidentified.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.21] collected minimizers\\n[M::mm_idx_gen::0.011*2.32] sorted minimizers\\n[M::main::0.011*2.32] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.20] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.12] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl060.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.asteroma_ccl060.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl060.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.24] collected minimizers\\n[M::mm_idx_gen::0.011*2.36] sorted minimizers\\n[M::main::0.011*2.36] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.23] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.15] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl068.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.asteroma_ccl068.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl068.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.28] collected minimizers\\n[M::mm_idx_gen::0.009*2.56] sorted minimizers\\n[M::main::0.009*2.55] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.38] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.011*2.28] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/saccharomyces_cerevisiae.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.saccharomyces_cerevisiae.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/saccharomyces_cerevisiae.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.011*1.16] collected minimizers\\n[M::mm_idx_gen::0.013*2.08] sorted minimizers\\n[M::main::0.013*2.07] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.015*1.99] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.015*1.93] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cladophialophora_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.cladophialophora_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cladophialophora_unidentified.15000.fasta\\n[M::main] Real time: 0.018 sec; CPU: 0.032 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.20] collected minimizers\\n[M::mm_idx_gen::0.012*2.18] sorted minimizers\\n[M::main::0.012*2.18] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.07] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*2.00] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_albicans.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_albicans.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_albicans.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.009*1.21] collected minimizers\\n[M::mm_idx_gen::0.012*2.21] sorted minimizers\\n[M::main::0.012*2.20] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.09] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*2.02] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_metapsilosis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_metapsilosis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_metapsilosis.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.21] collected minimizers\\n[M::mm_idx_gen::0.011*2.19] sorted minimizers\\n[M::main::0.011*2.19] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.11] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.06] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_orthopsilosis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_orthopsilosis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_orthopsilosis.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.19] collected minimizers\\n[M::mm_idx_gen::0.011*2.31] sorted minimizers\\n[M::main::0.011*2.31] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.21] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.013*2.14] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_parapsilosis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_parapsilosis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_parapsilosis.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.006*1.28] collected minimizers\\n[M::mm_idx_gen::0.009*2.39] sorted minimizers\\n[M::main::0.009*2.39] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.23] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.011*2.11] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_unidentified.15000.fasta\\n[M::main] Real time: 0.013 sec; CPU: 0.026 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.18] collected minimizers\\n[M::mm_idx_gen::0.012*1.97] sorted minimizers\\n[M::main::0.012*1.97] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.013*1.89] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.014*1.83] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_marxianus.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.kluyveromyces_marxianus.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_marxianus.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.23] collected minimizers\\n[M::mm_idx_gen::0.010*2.34] sorted minimizers\\n[M::main::0.010*2.34] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.21] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.13] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_kudriavzevii.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.pichia_kudriavzevii.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_kudriavzevii.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.25] collected minimizers\\n[M::mm_idx_gen::0.010*2.38] sorted minimizers\\n[M::main::0.010*2.38] loaded/built the index for 40 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.23] mid_occ = 41\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 40\\n[M::mm_idx_stat::0.012*2.13] distinct minimizers: 9451 (73.11% are singletons); average occurrences: 2.301; average spacing: 5.425\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_membranifaciens.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.pichia_membranifaciens.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.subdb.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_membranifaciens.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n"
     ]
    }
   ],
   "source": [
    "db_fn = sub_db_fn\n",
    "sub_db_mapping_fn = {}\n",
    "for species, fasta_fn in sub_reads_fn.items():\n",
    "    tmp_out = dbases_fn[db_fn]\n",
    "    db_name = os.path.basename(db_fn).replace('.fasta', '')\n",
    "    out_fn = os.path.join(tmp_out, F\"{db_name}.{species}.minimap2.paf\")\n",
    "    sub_db_mapping_fn[species] = out_fn\n",
    "    minimapmapping(fasta_fn, db_fn, out_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'puccinia_striiformis-tritici': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.puccinia_striiformis-tritici.minimap2.paf',\n",
       " 'zymoseptoria_tritici': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.zymoseptoria_tritici.minimap2.paf',\n",
       " 'pyrenophora_tritici-repentis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.pyrenophora_tritici-repentis.minimap2.paf',\n",
       " 'fusarium_oxysporum': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.fusarium_oxysporum.minimap2.paf',\n",
       " 'tuber_brumale': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.tuber_brumale.minimap2.paf',\n",
       " 'cortinarius_globuliformis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.cortinarius_globuliformis.minimap2.paf',\n",
       " 'aspergillus_niger': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.aspergillus_niger.minimap2.paf',\n",
       " 'clavispora_lusitaniae': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.clavispora_lusitaniae.minimap2.paf',\n",
       " 'kluyveromyces_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.kluyveromyces_unidentified.minimap2.paf',\n",
       " 'penicillium_chrysogenum': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.penicillium_chrysogenum.minimap2.paf',\n",
       " 'rhodotorula_mucilaginosa': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.rhodotorula_mucilaginosa.minimap2.paf',\n",
       " 'scedosporium_boydii': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.scedosporium_boydii.minimap2.paf',\n",
       " 'blastobotrys_proliferans': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.blastobotrys_proliferans.minimap2.paf',\n",
       " 'debaryomyces_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.debaryomyces_unidentified.minimap2.paf',\n",
       " 'galactomyces_geotrichum': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.galactomyces_geotrichum.minimap2.paf',\n",
       " 'kodamaea_ohmeri': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.kodamaea_ohmeri.minimap2.paf',\n",
       " 'meyerozyma_guillermondii': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.meyerozyma_guillermondii.minimap2.paf',\n",
       " 'wickerhamomyces_anomalus': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.wickerhamomyces_anomalus.minimap2.paf',\n",
       " 'yamadazyma_mexicana': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.yamadazyma_mexicana.minimap2.paf',\n",
       " 'yamadazyma_scolyti': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.yamadazyma_scolyti.minimap2.paf',\n",
       " 'yarrowia_lipolytica': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.yarrowia_lipolytica.minimap2.paf',\n",
       " 'zygoascus_hellenicus': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.zygoascus_hellenicus.minimap2.paf',\n",
       " 'aspergillus_flavus': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.aspergillus_flavus.minimap2.paf',\n",
       " 'cryptococcus_zero': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.cryptococcus_zero.minimap2.paf',\n",
       " 'aspergillus_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.aspergillus_unidentified.minimap2.paf',\n",
       " 'diaporthe_ccl067': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.diaporthe_ccl067.minimap2.paf',\n",
       " 'diaporthe_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.diaporthe_unidentified.minimap2.paf',\n",
       " 'oculimacula_yallundae-ccl031': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.oculimacula_yallundae-ccl031.minimap2.paf',\n",
       " 'oculimacula_yallundae-ccl029': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.oculimacula_yallundae-ccl029.minimap2.paf',\n",
       " 'dothiorella_vidmadera': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.dothiorella_vidmadera.minimap2.paf',\n",
       " 'quambalaria_cyanescens': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.quambalaria_cyanescens.minimap2.paf',\n",
       " 'entoleuca_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.entoleuca_unidentified.minimap2.paf',\n",
       " 'asteroma_ccl060': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.asteroma_ccl060.minimap2.paf',\n",
       " 'asteroma_ccl068': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.asteroma_ccl068.minimap2.paf',\n",
       " 'saccharomyces_cerevisiae': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.saccharomyces_cerevisiae.minimap2.paf',\n",
       " 'cladophialophora_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.cladophialophora_unidentified.minimap2.paf',\n",
       " 'candida_albicans': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_albicans.minimap2.paf',\n",
       " 'candida_metapsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_metapsilosis.minimap2.paf',\n",
       " 'candida_orthopsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_orthopsilosis.minimap2.paf',\n",
       " 'candida_parapsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_parapsilosis.minimap2.paf',\n",
       " 'candida_unidentified': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.candida_unidentified.minimap2.paf',\n",
       " 'kluyveromyces_marxianus': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.kluyveromyces_marxianus.minimap2.paf',\n",
       " 'pichia_kudriavzevii': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.pichia_kudriavzevii.minimap2.paf',\n",
       " 'pichia_membranifaciens': '/media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_subdb/gsref.subdb.pichia_membranifaciens.minimap2.paf'}"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_db_mapping_fn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.007*1.28] collected minimizers\\n[M::mm_idx_gen::0.009*2.48] sorted minimizers\\n[M::main::0.009*2.47] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.31] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.20] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/puccinia_striiformis-tritici.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.puccinia_striiformis-tritici.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/puccinia_striiformis-tritici.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.20] collected minimizers\\n[M::mm_idx_gen::0.012*2.20] sorted minimizers\\n[M::main::0.012*2.20] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.09] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*2.02] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zymoseptoria_tritici.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.zymoseptoria_tritici.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zymoseptoria_tritici.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.011*1.12] collected minimizers\\n[M::mm_idx_gen::0.014*1.99] sorted minimizers\\n[M::main::0.014*1.98] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.015*1.93] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.016*1.88] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pyrenophora_tritici-repentis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.pyrenophora_tritici-repentis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pyrenophora_tritici-repentis.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.16] collected minimizers\\n[M::mm_idx_gen::0.012*2.11] sorted minimizers\\n[M::main::0.012*2.11] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.02] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.96] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/fusarium_oxysporum.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.fusarium_oxysporum.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/fusarium_oxysporum.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.22] collected minimizers\\n[M::mm_idx_gen::0.010*2.41] sorted minimizers\\n[M::main::0.010*2.40] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.26] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.17] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/tuber_brumale.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.tuber_brumale.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/tuber_brumale.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.17] collected minimizers\\n[M::mm_idx_gen::0.012*2.28] sorted minimizers\\n[M::main::0.012*2.28] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.17] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*2.09] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cortinarius_globuliformis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.cortinarius_globuliformis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cortinarius_globuliformis.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.24] collected minimizers\\n[M::mm_idx_gen::0.010*2.44] sorted minimizers\\n[M::main::0.010*2.44] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.29] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.20] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_niger.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.aspergillus_niger.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_niger.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.13] collected minimizers\\n[M::mm_idx_gen::0.012*1.97] sorted minimizers\\n[M::main::0.012*1.97] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*1.90] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.85] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/clavispora_lusitaniae.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.clavispora_lusitaniae.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/clavispora_lusitaniae.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.18] collected minimizers\\n[M::mm_idx_gen::0.011*2.15] sorted minimizers\\n[M::main::0.011*2.15] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.05] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.013*1.98] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.kluyveromyces_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_unidentified.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.009*1.19] collected minimizers\\n[M::mm_idx_gen::0.012*2.32] sorted minimizers\\n[M::main::0.012*2.31] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*2.20] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*2.12] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/penicillium_chrysogenum.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.penicillium_chrysogenum.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/penicillium_chrysogenum.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.033 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.25] collected minimizers\\n[M::mm_idx_gen::0.010*2.32] sorted minimizers\\n[M::main::0.010*2.32] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.24] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.18] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/rhodotorula_mucilaginosa.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.rhodotorula_mucilaginosa.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/rhodotorula_mucilaginosa.15000.fasta\\n[M::main] Real time: 0.013 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.16] collected minimizers\\n[M::mm_idx_gen::0.012*2.09] sorted minimizers\\n[M::main::0.012*2.09] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.99] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.93] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/scedosporium_boydii.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.scedosporium_boydii.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/scedosporium_boydii.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.17] collected minimizers\\n[M::mm_idx_gen::0.013*2.18] sorted minimizers\\n[M::main::0.013*2.18] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*2.09] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*2.03] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/blastobotrys_proliferans.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.blastobotrys_proliferans.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/blastobotrys_proliferans.15000.fasta\\n[M::main] Real time: 0.018 sec; CPU: 0.034 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.21] collected minimizers\\n[M::mm_idx_gen::0.012*2.27] sorted minimizers\\n[M::main::0.012*2.27] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.18] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.013*2.11] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/debaryomyces_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.debaryomyces_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/debaryomyces_unidentified.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.12] collected minimizers\\n[M::mm_idx_gen::0.013*2.07] sorted minimizers\\n[M::main::0.013*2.07] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*2.01] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*1.98] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/galactomyces_geotrichum.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.galactomyces_geotrichum.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/galactomyces_geotrichum.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.23] collected minimizers\\n[M::mm_idx_gen::0.010*2.40] sorted minimizers\\n[M::main::0.010*2.40] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.26] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.17] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kodamaea_ohmeri.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.kodamaea_ohmeri.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kodamaea_ohmeri.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.17] collected minimizers\\n[M::mm_idx_gen::0.012*2.08] sorted minimizers\\n[M::main::0.012*2.08] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.98] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.92] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/meyerozyma_guillermondii.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.meyerozyma_guillermondii.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/meyerozyma_guillermondii.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.006*1.30] collected minimizers\\n[M::mm_idx_gen::0.008*2.54] sorted minimizers\\n[M::main::0.008*2.53] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.009*2.41] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.010*2.30] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/wickerhamomyces_anomalus.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.wickerhamomyces_anomalus.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/wickerhamomyces_anomalus.15000.fasta\\n[M::main] Real time: 0.012 sec; CPU: 0.024 sec; Peak RSS: 0.005 GB\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.008*1.20] collected minimizers\\n[M::mm_idx_gen::0.011*2.26] sorted minimizers\\n[M::main::0.011*2.25] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.15] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.013*2.08] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_mexicana.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.yamadazyma_mexicana.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_mexicana.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.16] collected minimizers\\n[M::mm_idx_gen::0.013*2.10] sorted minimizers\\n[M::main::0.013*2.10] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*2.03] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*1.98] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_scolyti.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.yamadazyma_scolyti.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yamadazyma_scolyti.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.17] collected minimizers\\n[M::mm_idx_gen::0.012*2.12] sorted minimizers\\n[M::main::0.012*2.11] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.04] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.99] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yarrowia_lipolytica.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.yarrowia_lipolytica.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/yarrowia_lipolytica.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.011*1.16] collected minimizers\\n[M::mm_idx_gen::0.014*2.12] sorted minimizers\\n[M::main::0.014*2.12] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*2.06] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*2.03] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zygoascus_hellenicus.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.zygoascus_hellenicus.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/zygoascus_hellenicus.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.011*1.16] collected minimizers\\n[M::mm_idx_gen::0.014*2.04] sorted minimizers\\n[M::main::0.014*2.04] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.015*1.98] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*1.96] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_flavus.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.aspergillus_flavus.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_flavus.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.22] collected minimizers\\n[M::mm_idx_gen::0.010*2.34] sorted minimizers\\n[M::main::0.010*2.34] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.22] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.13] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cryptococcus_zero.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.cryptococcus_zero.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cryptococcus_zero.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.027 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.24] collected minimizers\\n[M::mm_idx_gen::0.010*2.53] sorted minimizers\\n[M::main::0.010*2.52] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.37] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.27] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.aspergillus_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/aspergillus_unidentified.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.19] collected minimizers\\n[M::mm_idx_gen::0.012*2.20] sorted minimizers\\n[M::main::0.012*2.20] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.09] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*2.02] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_ccl067.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.diaporthe_ccl067.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_ccl067.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.23] collected minimizers\\n[M::mm_idx_gen::0.010*2.30] sorted minimizers\\n[M::main::0.010*2.30] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.17] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.08] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.diaporthe_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/diaporthe_unidentified.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.006*1.26] collected minimizers\\n[M::mm_idx_gen::0.009*2.60] sorted minimizers\\n[M::main::0.009*2.59] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.42] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.011*2.29] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl031.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.oculimacula_yallundae-ccl031.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl031.15000.fasta\\n[M::main] Real time: 0.013 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.006*1.28] collected minimizers\\n[M::mm_idx_gen::0.008*2.50] sorted minimizers\\n[M::main::0.008*2.50] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.33] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.010*2.21] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl029.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.oculimacula_yallundae-ccl029.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/oculimacula_yallundae-ccl029.15000.fasta\\n[M::main] Real time: 0.012 sec; CPU: 0.025 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.19] collected minimizers\\n[M::mm_idx_gen::0.012*2.09] sorted minimizers\\n[M::main::0.012*2.09] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.01] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.96] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/dothiorella_vidmadera.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.dothiorella_vidmadera.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/dothiorella_vidmadera.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.20] collected minimizers\\n[M::mm_idx_gen::0.012*2.23] sorted minimizers\\n[M::main::0.012*2.23] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.13] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*2.06] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/quambalaria_cyanescens.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.quambalaria_cyanescens.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/quambalaria_cyanescens.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.032 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.006*1.17] collected minimizers\\n[M::mm_idx_gen::0.009*2.64] sorted minimizers\\n[M::main::0.009*2.64] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.53] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.011*2.45] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/entoleuca_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.entoleuca_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/entoleuca_unidentified.15000.fasta\\n[M::main] Real time: 0.012 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.18] collected minimizers\\n[M::mm_idx_gen::0.011*2.31] sorted minimizers\\n[M::main::0.011*2.31] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.20] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.013*2.12] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl060.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.asteroma_ccl060.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl060.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.011*1.11] collected minimizers\\n[M::mm_idx_gen::0.014*2.03] sorted minimizers\\n[M::main::0.014*2.03] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.99] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*1.96] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl068.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.asteroma_ccl068.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/asteroma_ccl068.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.030 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.17] collected minimizers\\n[M::mm_idx_gen::0.010*2.49] sorted minimizers\\n[M::main::0.010*2.49] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*2.33] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.23] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/saccharomyces_cerevisiae.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.saccharomyces_cerevisiae.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/saccharomyces_cerevisiae.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.008*1.18] collected minimizers\\n[M::mm_idx_gen::0.011*2.36] sorted minimizers\\n[M::main::0.011*2.36] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.25] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.013*2.17] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cladophialophora_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.cladophialophora_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/cladophialophora_unidentified.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, \"[M::mm_idx_gen::0.011*1.13] collected minimizers\\n[M::mm_idx_gen::0.014*2.05] sorted minimizers\\n[M::main::0.014*2.05] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.015*1.98] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.016*1.94] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_albicans.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.candida_albicans.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_albicans.15000.fasta\\n[M::main] Real time: 0.018 sec; CPU: 0.032 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.011*1.11] collected minimizers\\n[M::mm_idx_gen::0.014*1.97] sorted minimizers\\n[M::main::0.014*1.97] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.015*1.90] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*1.85] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_metapsilosis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.candida_metapsilosis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_metapsilosis.15000.fasta\\n[M::main] Real time: 0.018 sec; CPU: 0.031 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.19] collected minimizers\\n[M::mm_idx_gen::0.011*2.08] sorted minimizers\\n[M::main::0.011*2.08] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.03] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*2.00] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_orthopsilosis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.candida_orthopsilosis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_orthopsilosis.15000.fasta\\n[M::main] Real time: 0.013 sec; CPU: 0.026 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.007*1.26] collected minimizers\\n[M::mm_idx_gen::0.009*2.51] sorted minimizers\\n[M::main::0.009*2.50] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.35] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.011*2.24] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_parapsilosis.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.candida_parapsilosis.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_parapsilosis.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.010*1.15] collected minimizers\\n[M::mm_idx_gen::0.012*1.97] sorted minimizers\\n[M::main::0.012*1.96] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*1.89] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.83] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_unidentified.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.candida_unidentified.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/candida_unidentified.15000.fasta\\n[M::main] Real time: 0.017 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.19] collected minimizers\\n[M::mm_idx_gen::0.012*2.10] sorted minimizers\\n[M::main::0.012*2.10] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.01] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.95] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_marxianus.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.kluyveromyces_marxianus.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/kluyveromyces_marxianus.15000.fasta\\n[M::main] Real time: 0.016 sec; CPU: 0.029 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.17] collected minimizers\\n[M::mm_idx_gen::0.011*2.12] sorted minimizers\\n[M::main::0.011*2.12] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.03] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.013*1.97] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_kudriavzevii.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.pichia_kudriavzevii.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_kudriavzevii.15000.fasta\\n[M::main] Real time: 0.015 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n",
      "(0, \"[M::mm_idx_gen::0.009*1.14] collected minimizers\\n[M::mm_idx_gen::0.011*2.18] sorted minimizers\\n[M::main::0.011*2.18] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.012*2.11] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.013*2.05] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\nERROR: failed to open file '/media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_membranifaciens.15000.fasta'\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref_db/gsref.db.pichia_membranifaciens.minimap2.paf /media/MassStorage/tmp/TE/honours/analysis/Mapping/gsref.db.fasta /media/MassStorage/tmp/TE/honours/analysis/Mapping/MC_READS/pichia_membranifaciens.15000.fasta\\n[M::main] Real time: 0.014 sec; CPU: 0.028 sec; Peak RSS: 0.005 GB\")\n"
     ]
    }
   ],
   "source": [
    "db_fn = new_db_fn\n",
    "new_db_mapping_fn = {}\n",
    "for species, fasta_fn in sub_reads_fn.items():\n",
    "    tmp_out = dbases_fn[db_fn]\n",
    "    db_name = os.path.basename(db_fn).replace('.fasta', '')\n",
    "    out_fn = os.path.join(tmp_out, F\"{db_name}.{species}.minimap2.paf\")\n",
    "    new_db_mapping_fn[species] = out_fn\n",
    "    minimapmapping(fasta_fn, db_fn, out_fn)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Look at mapping results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mapping_results(fn, species):\n",
    "    min_header = ['qseqid', 'qlen', 'qstart', 'qstop', 'strand', 'tname', 'tlen', 'tstart', 'tend', 'nmatch', 'alen', 'mquality']\n",
    "    tmp_df = pd.read_csv(fn, sep='\\t', header = None, usecols=[x for x in range(0,12)], names=min_header)\n",
    "    sub_df = tmp_df[tmp_df['mquality'] == tmp_df.groupby('qseqid')['mquality'].transform(max)].reset_index(drop=True)\n",
    "    sub_df = sub_df[tmp_df['nmatch'] == tmp_df.groupby('qseqid')['nmatch'].transform(max)].reset_index(drop=True)\n",
    "    hit_series = pd.Series(sub_df.groupby('tname')['mquality'].count().tolist()/sub_df.groupby('tname')['mquality'].count().sum(),\n",
    "                      sub_df.groupby('tname')['mquality'].count().index)\n",
    "    hit_series.sort_values(ascending=False, inplace=True)\n",
    "    print('##########\\n')\n",
    "    print(F\"This was the query species: {species}\\n\")\n",
    "    print(F\"These are the results:\")\n",
    "    print(hit_series,'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##########\n",
      "\n",
      "This was the query species: puccinia_striiformis-tritici\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: zymoseptoria_tritici\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: pyrenophora_tritici-repentis\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: fusarium_oxysporum\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: tuber_brumale\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: cortinarius_globuliformis\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: aspergillus_niger\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: clavispora_lusitaniae\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: kluyveromyces_unidentified\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: penicillium_chrysogenum\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: rhodotorula_mucilaginosa\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: scedosporium_boydii\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: blastobotrys_proliferans\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: debaryomyces_unidentified\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: galactomyces_geotrichum\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: kodamaea_ohmeri\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: meyerozyma_guillermondii\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: wickerhamomyces_anomalus\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: yamadazyma_mexicana\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: yamadazyma_scolyti\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: yarrowia_lipolytica\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: zygoascus_hellenicus\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: aspergillus_flavus\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: cryptococcus_zero\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: aspergillus_unidentified\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: diaporthe_ccl067\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: diaporthe_unidentified\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: oculimacula_yallundae-ccl031\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: oculimacula_yallundae-ccl029\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: dothiorella_vidmadera\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: quambalaria_cyanescens\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: entoleuca_unidentified\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: asteroma_ccl060\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: asteroma_ccl068\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: saccharomyces_cerevisiae\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: cladophialophora_unidentified\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_albicans\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_metapsilosis\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_orthopsilosis\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_parapsilosis\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_unidentified\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: kluyveromyces_marxianus\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: pichia_kudriavzevii\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: pichia_membranifaciens\n",
      "\n",
      "These are the results:\n",
      "Series([], dtype: float64) \n",
      "\n"
     ]
    }
   ],
   "source": [
    "###this is running the reads against the full database\n",
    "for species, hit_fn in new_db_mapping_fn.items():\n",
    "    mapping_results(hit_fn, species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "###this is running against a database that have ['Candida_orthopsilosis', 'Candida_metapsilosis', 'Aspergillus_niger'] deleted\n",
    "for species, hit_fn in sub_db_mapping_fn.items():\n",
    "    mapping_results(hit_fn, species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
