{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Notebook to analyzse the efficiency of minimap mapping against a mock community"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Starting points from Tavish  \n",
    "reference_dataframe at '/media/MassStorage/tmp/TE/honours/analysis/Stats/reference_dataframe.csv'  \n",
    "custom_database at '/media/MassStorage/tmp/TE/honours/database/custom_database_labelled.fasta'  \n",
    "taxonomy_file at '/media/MassStorage/tmp/TE/honours/analysis/Stats/taxonomy_file.csv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### workflow\n",
    "\n",
    "* Two databases\n",
    "* subsample 15000 reads per each mock community species. Save those out.\n",
    "* map reads against both databases with minimap safe out data in paf format.\n",
    "* get best hit per species (see what this means while looking at the data).\n",
    "* add the full taxonomy to each best match using the taxonomy file.\n",
    "* summarize data at different taxonomic ranks for each species.\n",
    "* pull this all together somehow (summary across all the samples? focus on species of interest e.g. deleted from analyis?)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import os\n",
    "import random\n",
    "import subprocess\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Initial data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "reference_dataframe_fn = os.path.abspath('/media/MassStorage/tmp/TE/honours/analysis/Stats/reference_dataframe.csv')\n",
    "max_custom_database_fn = os.path.abspath('/media/MassStorage/tmp/TE/honours/database/custom_database_labelled.fasta')\n",
    "taxonomy_file_fn = os.path.abspath('/media/MassStorage/tmp/TE/honours/analysis/Stats/taxonomy_file.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "INPUT_BASEDIR = os.path.abspath('/media/MassStorage/tmp/TE/honours')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "OUT_DIR = os.path.abspath('../../analysis/Mapping_mock_gsref')\n",
    "if not os.path.exists(OUT_DIR):\n",
    "    os.mkdir(OUT_DIR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "### list of species in the max database\n",
    "max_species = ['Puccinia_striiformis_tritici',\n",
    " 'Zymoseptoria_tritici',\n",
    " 'Pyrenophora_tritici-repentis',\n",
    " 'Fusarium_oxysporum',\n",
    " 'Tuber_brumale',\n",
    " 'Cortinarius_globuliformis',\n",
    " 'Aspergillus_niger',\n",
    " 'Clavispora_lusitaniae',\n",
    " 'Cryptococcus_neoformans',\n",
    " 'Penicillium_chrysogenum',\n",
    " 'Rhodotorula_mucilaginosa',\n",
    " 'Scedosporium_boydii',\n",
    " 'Blastobotrys_proliferans',\n",
    " 'Candida_zeylanoides',\n",
    " 'Galactomyces_geotrichum',\n",
    " 'Kodamaea_ohmeri',\n",
    " 'Meyerozyma_guillermondii',\n",
    " 'Wickerhamomyces_anomalus',\n",
    " 'Yamadazyma_mexicana',\n",
    " 'Yamadazyma_scolyti',\n",
    " 'Yarrowia_lipolytica',\n",
    " 'Zygoascus_hellenicus',\n",
    " 'Aspergillus_flavus',\n",
    " 'Cryptococcus_zero',\n",
    " 'Aspergillus_sp.',\n",
    " 'CCL067',\n",
    " 'Diaporthe_sp.',\n",
    " 'Tapesia_yallundae_CCL031',\n",
    " 'Tapesia_yallundae_CCL029',\n",
    " 'Dothiorella_vidmadera',\n",
    " 'Quambalaria_cyanescens',\n",
    " 'Entoleuca_sp.',\n",
    " 'CCL060',\n",
    " 'CCL068',\n",
    " 'Saccharomyces_cerevisiae',\n",
    " 'Cladophialophora_sp.',\n",
    " 'Candida_albicans',\n",
    " 'Candida_metapsilosis',\n",
    " 'Candida_orthopsilosis',\n",
    " 'Candida_parapsilosis',\n",
    " 'Geotrichum_candidum',\n",
    " 'Kluyveromyces_lactis',\n",
    " 'Kluyveromyces_marxianus',\n",
    " 'Pichia_kudriavzevii',\n",
    " 'Pichia_membranifaciens']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "###Removed from second test databes\n",
    "species_delete = ['Candida_orthopsilosis',\n",
    "                 'Candida_metapsilosis',\n",
    "                 'Aspergillus_niger']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "###species to be searched against both databases\n",
    "mock_community = ['Penicillium_chrysogenum',\n",
    " 'Aspergillus_flavus',\n",
    " 'Aspergillus_niger',\n",
    " 'Pichia_kudriavzevii',\n",
    " 'Pichia_membranifaciens',\n",
    " 'Candida_albicans',\n",
    " 'Candida_parapsilosis',\n",
    " 'Candida_orthopsilosis',\n",
    " 'Candida_metapsilosis']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "fixed_old_names = ['Kluyveromyces_lactis',\n",
    "                   'Candida_zeylanoides',\n",
    "                   'Cladophialophora_sp.',\n",
    "                   'Diaporthe_sp.',\n",
    "                   'CCL060',\n",
    "                   'CCL068',\n",
    "                   'CCL067',\n",
    "                   'Aspergillus_sp.',\n",
    "                   'Entoleuca_sp.',\n",
    "                   'Tapesia_yallundae_CCL029',\n",
    "                   'Tapesia_yallundae_CCL031',\n",
    "                   'Cryptococcus_neoformans']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "fixed_new_names = ['candida_unidentified',\n",
    "                   'debaryomyces_unidentified',\n",
    "                   'cladophialophora_unidentified',\n",
    "                   'diaporthe_unidentified',\n",
    "                   'asteroma_ccl060',\n",
    "                   'asteroma_ccl068',\n",
    "                   'diaporthe_ccl067',\n",
    "                   'aspergillus_unidentified',\n",
    "                   'entoleuca_unidentified',\n",
    "                   'oculimacula_yallundae-ccl029',\n",
    "                   'oculimacula_yallundae-ccl031',\n",
    "                   'kluyveromyces_unidentified']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "old_to_new_names = dict(zip(fixed_old_names, fixed_new_names))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Kluyveromyces_lactis': 'candida_unidentified',\n",
       " 'Candida_zeylanoides': 'debaryomyces_unidentified',\n",
       " 'Cladophialophora_sp.': 'cladophialophora_unidentified',\n",
       " 'Diaporthe_sp.': 'diaporthe_unidentified',\n",
       " 'CCL060': 'asteroma_ccl060',\n",
       " 'CCL068': 'asteroma_ccl068',\n",
       " 'CCL067': 'diaporthe_ccl067',\n",
       " 'Aspergillus_sp.': 'aspergillus_unidentified',\n",
       " 'Entoleuca_sp.': 'entoleuca_unidentified',\n",
       " 'Tapesia_yallundae_CCL029': 'oculimacula_yallundae-ccl029',\n",
       " 'Tapesia_yallundae_CCL031': 'oculimacula_yallundae-ccl031',\n",
       " 'Cryptococcus_neoformans': 'kluyveromyces_unidentified'}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_to_new_names"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Fix databases and names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_df = pd.read_csv(reference_dataframe_fn)\n",
    "ref_df['name_species'] = ref_df['genus'] +\"_\"+ ref_df['species']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['puccinia_striiformis-tritici',\n",
       " 'zymoseptoria_tritici',\n",
       " 'pyrenophora_tritici-repentis',\n",
       " 'fusarium_oxysporum',\n",
       " 'tuber_brumale',\n",
       " 'cortinarius_globuliformis',\n",
       " 'aspergillus_niger',\n",
       " 'clavispora_lusitaniae',\n",
       " 'kluyveromyces_unidentified',\n",
       " 'penicillium_chrysogenum',\n",
       " 'rhodotorula_mucilaginosa',\n",
       " 'scedosporium_boydii',\n",
       " 'blastobotrys_proliferans',\n",
       " 'debaryomyces_unidentified',\n",
       " 'galactomyces_geotrichum',\n",
       " 'kodamaea_ohmeri',\n",
       " 'meyerozyma_guillermondii',\n",
       " 'wickerhamomyces_anomalus',\n",
       " 'yamadazyma_mexicana',\n",
       " 'yamadazyma_scolyti',\n",
       " 'yarrowia_lipolytica',\n",
       " 'zygoascus_hellenicus',\n",
       " 'aspergillus_flavus',\n",
       " 'cryptococcus_zero',\n",
       " 'aspergillus_unidentified',\n",
       " 'diaporthe_ccl067',\n",
       " 'diaporthe_unidentified',\n",
       " 'oculimacula_yallundae-ccl031',\n",
       " 'oculimacula_yallundae-ccl029',\n",
       " 'dothiorella_vidmadera',\n",
       " 'quambalaria_cyanescens',\n",
       " 'entoleuca_unidentified',\n",
       " 'asteroma_ccl060',\n",
       " 'asteroma_ccl068',\n",
       " 'saccharomyces_cerevisiae',\n",
       " 'cladophialophora_unidentified',\n",
       " 'candida_albicans',\n",
       " 'candida_metapsilosis',\n",
       " 'candida_orthopsilosis',\n",
       " 'candida_parapsilosis',\n",
       " 'candida_unidentified',\n",
       " 'kluyveromyces_marxianus',\n",
       " 'pichia_kudriavzevii',\n",
       " 'pichia_membranifaciens']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_df.name_species.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_db_fn = os.path.join(OUT_DIR, 'gsref.db.fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Cryptococcus_gattii\n",
      "Geotrichum_candidum\n"
     ]
    }
   ],
   "source": [
    "new_db_list = []\n",
    "old_db_list = []\n",
    "for seq in SeqIO.parse(max_custom_database_fn, 'fasta'):\n",
    "    old_db_list.append(seq.id)\n",
    "    if seq.id in old_to_new_names.keys():\n",
    "        #print(seq.id)\n",
    "        seq.id = seq.name = seq.description = old_to_new_names[seq.id]\n",
    "        new_db_list.append(seq)\n",
    "    elif seq.id.lower() in ref_df.name_species.tolist():\n",
    "        #print(seq.id)\n",
    "        seq.id = seq.name = seq.description = seq.id.lower()\n",
    "        new_db_list.append(seq)\n",
    "    else:\n",
    "        print(seq.id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "if len(new_db_list) == len(old_db_list) -2:\n",
    "    SeqIO.write(new_db_list, new_db_fn, 'fasta')\n",
    "else:\n",
    "    print(\"please check!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "sub_db_fn = os.path.join(OUT_DIR, 'gsref.subdb.fasta')\n",
    "sub_db_list = []\n",
    "for seq in new_db_list:\n",
    "    if seq.id not in [x.lower() for x in species_delete]:\n",
    "        sub_db_list.append(seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "if len(sub_db_list) + len(species_delete) == len(new_db_list):\n",
    "    SeqIO.write(sub_db_list, sub_db_fn, 'fasta' )\n",
    "else:\n",
    "    print(\"please check!\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['puccinia_striiformis-tritici',\n",
       " 'zymoseptoria_tritici',\n",
       " 'pyrenophora_tritici-repentis',\n",
       " 'fusarium_oxysporum',\n",
       " 'tuber_brumale',\n",
       " 'cortinarius_globuliformis',\n",
       " 'clavispora_lusitaniae',\n",
       " 'kluyveromyces_unidentified',\n",
       " 'penicillium_chrysogenum',\n",
       " 'rhodotorula_mucilaginosa',\n",
       " 'scedosporium_boydii',\n",
       " 'blastobotrys_proliferans',\n",
       " 'debaryomyces_unidentified',\n",
       " 'galactomyces_geotrichum',\n",
       " 'kodamaea_ohmeri',\n",
       " 'meyerozyma_guillermondii',\n",
       " 'wickerhamomyces_anomalus',\n",
       " 'yamadazyma_mexicana',\n",
       " 'yamadazyma_scolyti',\n",
       " 'yarrowia_lipolytica',\n",
       " 'zygoascus_hellenicus',\n",
       " 'aspergillus_flavus',\n",
       " 'cryptococcus_zero',\n",
       " 'aspergillus_unidentified',\n",
       " 'diaporthe_ccl067',\n",
       " 'diaporthe_unidentified',\n",
       " 'oculimacula_yallundae-ccl031',\n",
       " 'oculimacula_yallundae-ccl029',\n",
       " 'dothiorella_vidmadera',\n",
       " 'quambalaria_cyanescens',\n",
       " 'entoleuca_unidentified',\n",
       " 'asteroma_ccl060',\n",
       " 'asteroma_ccl068',\n",
       " 'saccharomyces_cerevisiae',\n",
       " 'cladophialophora_unidentified',\n",
       " 'candida_albicans',\n",
       " 'candida_parapsilosis',\n",
       " 'candida_unidentified',\n",
       " 'kluyveromyces_marxianus',\n",
       " 'pichia_kudriavzevii',\n",
       " 'pichia_membranifaciens']"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x.id for x in sub_db_list]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "mock_community = [x.lower() for x in mock_community]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['penicillium_chrysogenum',\n",
       " 'aspergillus_flavus',\n",
       " 'aspergillus_niger',\n",
       " 'pichia_kudriavzevii',\n",
       " 'pichia_membranifaciens',\n",
       " 'candida_albicans',\n",
       " 'candida_parapsilosis',\n",
       " 'candida_orthopsilosis',\n",
       " 'candida_metapsilosis']"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mock_community"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Subsample reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "def subsamplereads(in_fn, out_fn, n_reads):\n",
    "    command = F'reformat.sh samplereadstarget={n_reads} in={in_fn} out={out_fn}'\n",
    "    out = subprocess.getstatusoutput(command)\n",
    "    if out[0] == 1:\n",
    "        print(F\":)Completed {command}\\n\")\n",
    "    else:\n",
    "        print(F\":(check one {command}!!\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "n_reads = 15000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "MC_READ_DIR = os.path.join(OUT_DIR, 'MC_READS')\n",
    "if not os.path.exists(MC_READ_DIR):\n",
    "    os.mkdir(MC_READ_DIR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'species', 'genus', 'family', 'order', 'class', 'phylum',\n",
       "       'kingdom', '# raw reads', '# reads after homology filtering',\n",
       "       '# reads after length filtering', '# for use', 'path to raw reads',\n",
       "       'path to homology filtering', 'path to length filtering',\n",
       "       'path for use', 'name_species'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ref_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'penicillium_chrysogenum': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode10/length_restricted_for_use.fasta',\n",
       " 'aspergillus_flavus': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode12/length_restricted_for_use.fasta',\n",
       " 'aspergillus_niger': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode07/length_restricted_for_use.fasta',\n",
       " 'pichia_kudriavzevii': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode11/length_restricted_for_use.fasta',\n",
       " 'pichia_membranifaciens': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode12/length_restricted_for_use.fasta',\n",
       " 'candida_albicans': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode03/length_restricted_for_use.fasta',\n",
       " 'candida_parapsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode06/length_restricted_for_use.fasta',\n",
       " 'candida_orthopsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode05/length_restricted_for_use.fasta',\n",
       " 'candida_metapsilosis': '/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode04/length_restricted_for_use.fasta'}"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fn_subsampling = {}\n",
    "for x in mock_community:\n",
    "    fn_subsampling[x] = (ref_df[(ref_df['species'] == x.split('_')[1]) & (ref_df['genus'] == x.split('_')[0])]['path for use'].tolist()[0])\n",
    "    fn_subsampling[x] = os.path.join(INPUT_BASEDIR, fn_subsampling[x])\n",
    "fn_subsampling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode10/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/penicillium_chrysogenum.15000.fasta\n",
      "\n",
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171207_FAH18654/barcode12/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/aspergillus_flavus.15000.fasta\n",
      "\n",
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20171103_FAH15473/barcode07/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/aspergillus_niger.15000.fasta\n",
      "\n",
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode11/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/pichia_kudriavzevii.15000.fasta\n",
      "\n",
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode12/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/pichia_membranifaciens.15000.fasta\n",
      "\n",
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode03/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_albicans.15000.fasta\n",
      "\n",
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode06/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_parapsilosis.15000.fasta\n",
      "\n",
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode05/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_orthopsilosis.15000.fasta\n",
      "\n",
      ":)Completed reformat.sh samplereadstarget=15000 in=/media/MassStorage/tmp/TE/honours/analysis/Length_Filtered/20180108_FAH18647/barcode04/length_restricted_for_use.fasta out=/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_metapsilosis.15000.fasta\n",
      "\n"
     ]
    }
   ],
   "source": [
    "sub_reads_fn = {}\n",
    "for key, value in fn_subsampling.items():\n",
    "    species = key\n",
    "    in_fn = value\n",
    "    out_fn = os.path.join(MC_READ_DIR, F'{species}.{n_reads}.fasta')\n",
    "    subsamplereads(in_fn, out_fn, n_reads)\n",
    "    sub_reads_fn[species] = out_fn"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Map with minimap against both databases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "def minimapmapping(fasta_fn, ref_fn, out_fn):\n",
    "    command = F\"minimap2 -x map-ont -t 6 {ref_fn} {fasta_fn} -o {out_fn}\"\n",
    "    out = subprocess.getstatusoutput(command)\n",
    "    print(out)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb',\n",
       " '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db'}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dbases_fn = {}\n",
    "for x in [sub_db_fn, new_db_fn]:\n",
    "    dbases_fn[x] = os.path.join(OUT_DIR, os.path.basename(x).replace('.fasta', '').replace('.','_'))\n",
    "    if not os.path.exists(dbases_fn[x]):\n",
    "        os.mkdir(dbases_fn[x])\n",
    "dbases_fn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, '[M::mm_idx_gen::0.009*1.19] collected minimizers\\n[M::mm_idx_gen::0.012*2.11] sorted minimizers\\n[M::main::0.012*2.10] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.013*2.01] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.014*1.94] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::0.982*5.32] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.penicillium_chrysogenum.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/penicillium_chrysogenum.15000.fasta\\n[M::main] Real time: 0.983 sec; CPU: 5.226 sec; Peak RSS: 0.094 GB')\n",
      "(0, '[M::mm_idx_gen::0.007*1.17] collected minimizers\\n[M::mm_idx_gen::0.009*2.08] sorted minimizers\\n[M::main::0.009*2.08] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.010*1.98] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.010*1.91] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::1.126*5.46] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.aspergillus_flavus.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/aspergillus_flavus.15000.fasta\\n[M::main] Real time: 1.127 sec; CPU: 6.152 sec; Peak RSS: 0.095 GB')\n",
      "(0, '[M::mm_idx_gen::0.012*1.15] collected minimizers\\n[M::mm_idx_gen::0.014*1.91] sorted minimizers\\n[M::main::0.014*1.91] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.015*1.86] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.016*1.82] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::0.949*5.29] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.aspergillus_niger.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/aspergillus_niger.15000.fasta\\n[M::main] Real time: 0.951 sec; CPU: 5.022 sec; Peak RSS: 0.094 GB')\n",
      "(0, '[M::mm_idx_gen::0.009*1.14] collected minimizers\\n[M::mm_idx_gen::0.011*1.84] sorted minimizers\\n[M::main::0.011*1.84] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.012*1.77] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.013*1.73] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::0.947*5.35] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.pichia_kudriavzevii.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/pichia_kudriavzevii.15000.fasta\\n[M::main] Real time: 0.948 sec; CPU: 5.069 sec; Peak RSS: 0.092 GB')\n",
      "(0, '[M::mm_idx_gen::0.006*1.19] collected minimizers\\n[M::mm_idx_gen::0.008*2.13] sorted minimizers\\n[M::main::0.008*2.13] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.009*2.01] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.010*1.93] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::0.930*5.32] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.pichia_membranifaciens.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/pichia_membranifaciens.15000.fasta\\n[M::main] Real time: 0.932 sec; CPU: 4.951 sec; Peak RSS: 0.090 GB')\n",
      "(0, '[M::mm_idx_gen::0.007*1.14] collected minimizers\\n[M::mm_idx_gen::0.010*2.14] sorted minimizers\\n[M::main::0.010*2.13] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.04] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.011*1.98] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::1.170*5.45] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.candida_albicans.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_albicans.15000.fasta\\n[M::main] Real time: 1.171 sec; CPU: 6.380 sec; Peak RSS: 0.092 GB')\n",
      "(0, '[M::mm_idx_gen::0.009*1.16] collected minimizers\\n[M::mm_idx_gen::0.011*2.02] sorted minimizers\\n[M::main::0.011*2.02] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.012*1.96] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.012*1.92] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::1.228*5.47] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.candida_parapsilosis.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_parapsilosis.15000.fasta\\n[M::main] Real time: 1.230 sec; CPU: 6.725 sec; Peak RSS: 0.093 GB')\n",
      "(0, '[M::mm_idx_gen::0.009*1.14] collected minimizers\\n[M::mm_idx_gen::0.011*1.79] sorted minimizers\\n[M::main::0.011*1.79] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.012*1.73] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.013*1.68] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::1.209*5.44] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.candida_orthopsilosis.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_orthopsilosis.15000.fasta\\n[M::main] Real time: 1.210 sec; CPU: 6.576 sec; Peak RSS: 0.092 GB')\n",
      "(0, '[M::mm_idx_gen::0.011*1.15] collected minimizers\\n[M::mm_idx_gen::0.013*1.92] sorted minimizers\\n[M::main::0.013*1.92] loaded/built the index for 41 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.85] mid_occ = 42\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 41\\n[M::mm_idx_stat::0.015*1.80] distinct minimizers: 9557 (72.65% are singletons); average occurrences: 2.332; average spacing: 5.427\\n[M::worker_pipeline::1.214*5.46] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.candida_metapsilosis.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.subdb.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_metapsilosis.15000.fasta\\n[M::main] Real time: 1.215 sec; CPU: 6.623 sec; Peak RSS: 0.093 GB')\n"
     ]
    }
   ],
   "source": [
    "db_fn = sub_db_fn\n",
    "sub_db_mapping_fn = {}\n",
    "for species, fasta_fn in sub_reads_fn.items():\n",
    "    tmp_out = dbases_fn[db_fn]\n",
    "    db_name = os.path.basename(db_fn).replace('.fasta', '')\n",
    "    out_fn = os.path.join(tmp_out, F\"{db_name}.{species}.minimap2.paf\")\n",
    "    sub_db_mapping_fn[species] = out_fn\n",
    "    minimapmapping(fasta_fn, db_fn, out_fn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'penicillium_chrysogenum': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.penicillium_chrysogenum.minimap2.paf',\n",
       " 'aspergillus_flavus': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.aspergillus_flavus.minimap2.paf',\n",
       " 'aspergillus_niger': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.aspergillus_niger.minimap2.paf',\n",
       " 'pichia_kudriavzevii': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.pichia_kudriavzevii.minimap2.paf',\n",
       " 'pichia_membranifaciens': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.pichia_membranifaciens.minimap2.paf',\n",
       " 'candida_albicans': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.candida_albicans.minimap2.paf',\n",
       " 'candida_parapsilosis': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.candida_parapsilosis.minimap2.paf',\n",
       " 'candida_orthopsilosis': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.candida_orthopsilosis.minimap2.paf',\n",
       " 'candida_metapsilosis': '/media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_subdb/gsref.subdb.candida_metapsilosis.minimap2.paf'}"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_db_mapping_fn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(0, '[M::mm_idx_gen::0.005*1.20] collected minimizers\\n[M::mm_idx_gen::0.007*2.30] sorted minimizers\\n[M::main::0.007*2.30] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.008*2.18] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.009*2.10] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::1.029*5.38] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.penicillium_chrysogenum.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/penicillium_chrysogenum.15000.fasta\\n[M::main] Real time: 1.031 sec; CPU: 5.542 sec; Peak RSS: 0.098 GB')\n",
      "(0, '[M::mm_idx_gen::0.008*1.15] collected minimizers\\n[M::mm_idx_gen::0.010*2.06] sorted minimizers\\n[M::main::0.010*2.06] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.010*1.98] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.011*1.92] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::1.239*5.46] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.aspergillus_flavus.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/aspergillus_flavus.15000.fasta\\n[M::main] Real time: 1.241 sec; CPU: 6.774 sec; Peak RSS: 0.098 GB')\n",
      "(0, '[M::mm_idx_gen::0.009*1.12] collected minimizers\\n[M::mm_idx_gen::0.012*2.05] sorted minimizers\\n[M::main::0.012*2.05] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.013*1.98] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.92] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::1.047*5.32] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.aspergillus_niger.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/aspergillus_niger.15000.fasta\\n[M::main] Real time: 1.048 sec; CPU: 5.566 sec; Peak RSS: 0.098 GB')\n",
      "(0, '[M::mm_idx_gen::0.009*1.13] collected minimizers\\n[M::mm_idx_gen::0.010*1.86] sorted minimizers\\n[M::main::0.011*1.86] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*1.80] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*1.75] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::1.008*5.38] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.pichia_kudriavzevii.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/pichia_kudriavzevii.15000.fasta\\n[M::main] Real time: 1.010 sec; CPU: 5.425 sec; Peak RSS: 0.096 GB')\n",
      "(0, '[M::mm_idx_gen::0.009*1.10] collected minimizers\\n[M::mm_idx_gen::0.011*1.82] sorted minimizers\\n[M::main::0.011*1.82] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.011*1.76] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.012*1.72] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::0.978*5.36] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.pichia_membranifaciens.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/pichia_membranifaciens.15000.fasta\\n[M::main] Real time: 0.980 sec; CPU: 5.246 sec; Peak RSS: 0.094 GB')\n",
      "(0, '[M::mm_idx_gen::0.007*1.19] collected minimizers\\n[M::mm_idx_gen::0.009*2.13] sorted minimizers\\n[M::main::0.009*2.12] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.010*2.03] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.011*1.96] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::1.269*5.46] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.candida_albicans.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_albicans.15000.fasta\\n[M::main] Real time: 1.271 sec; CPU: 6.932 sec; Peak RSS: 0.095 GB')\n",
      "(0, '[M::mm_idx_gen::0.010*1.14] collected minimizers\\n[M::mm_idx_gen::0.013*1.93] sorted minimizers\\n[M::main::0.013*1.93] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.86] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.81] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::1.381*5.48] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.candida_parapsilosis.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_parapsilosis.15000.fasta\\n[M::main] Real time: 1.382 sec; CPU: 7.562 sec; Peak RSS: 0.096 GB')\n",
      "(0, '[M::mm_idx_gen::0.011*1.13] collected minimizers\\n[M::mm_idx_gen::0.013*1.82] sorted minimizers\\n[M::main::0.013*1.82] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.77] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.014*1.73] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::1.347*5.48] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.candida_orthopsilosis.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_orthopsilosis.15000.fasta\\n[M::main] Real time: 1.349 sec; CPU: 7.383 sec; Peak RSS: 0.096 GB')\n",
      "(0, '[M::mm_idx_gen::0.011*1.11] collected minimizers\\n[M::mm_idx_gen::0.013*1.82] sorted minimizers\\n[M::main::0.013*1.81] loaded/built the index for 44 target sequence(s)\\n[M::mm_mapopt_update::0.014*1.76] mid_occ = 45\\n[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 44\\n[M::mm_idx_stat::0.015*1.72] distinct minimizers: 9696 (71.01% are singletons); average occurrences: 2.462; average spacing: 5.428\\n[M::worker_pipeline::1.372*5.47] mapped 15000 sequences\\n[M::main] Version: 2.17-r941\\n[M::main] CMD: minimap2 -x map-ont -t 6 -o /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref_db/gsref.db.candida_metapsilosis.minimap2.paf /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/gsref.db.fasta /media/WorkingStorage/ben.working/students/tavish/analysis/Mapping_mock_gsref/MC_READS/candida_metapsilosis.15000.fasta\\n[M::main] Real time: 1.374 sec; CPU: 7.509 sec; Peak RSS: 0.096 GB')\n"
     ]
    }
   ],
   "source": [
    "db_fn = new_db_fn\n",
    "new_db_mapping_fn = {}\n",
    "for species, fasta_fn in sub_reads_fn.items():\n",
    "    tmp_out = dbases_fn[db_fn]\n",
    "    db_name = os.path.basename(db_fn).replace('.fasta', '')\n",
    "    out_fn = os.path.join(tmp_out, F\"{db_name}.{species}.minimap2.paf\")\n",
    "    new_db_mapping_fn[species] = out_fn\n",
    "    minimapmapping(fasta_fn, db_fn, out_fn)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Look at mapping results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "def mapping_results(fn, species):\n",
    "    min_header = ['qseqid', 'qlen', 'qstart', 'qstop', 'strand', 'tname', 'tlen', 'tstart', 'tend', 'nmatch', 'alen', 'mquality']\n",
    "    tmp_df = pd.read_csv(fn, sep='\\t', header = None, usecols=[x for x in range(0,12)], names=min_header)\n",
    "    sub_df = tmp_df[tmp_df['mquality'] == tmp_df.groupby('qseqid')['mquality'].transform(max)].reset_index(drop=True)\n",
    "    sub_df = sub_df[tmp_df['nmatch'] == tmp_df.groupby('qseqid')['nmatch'].transform(max)].reset_index(drop=True)\n",
    "    hit_series = pd.Series(sub_df.groupby('tname')['mquality'].count().tolist()/sub_df.groupby('tname')['mquality'].count().sum(),\n",
    "                      sub_df.groupby('tname')['mquality'].count().index)\n",
    "    hit_series.sort_values(ascending=False, inplace=True)\n",
    "    print('##########\\n')\n",
    "    print(F\"This was the query species: {species}\\n\")\n",
    "    print(F\"These are the results:\")\n",
    "    print(hit_series,'\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##########\n",
      "\n",
      "This was the query species: penicillium_chrysogenum\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "penicillium_chrysogenum          0.987957\n",
      "aspergillus_niger                0.002348\n",
      "aspergillus_flavus               0.001894\n",
      "aspergillus_unidentified         0.001515\n",
      "tuber_brumale                    0.000682\n",
      "kluyveromyces_unidentified       0.000606\n",
      "kluyveromyces_marxianus          0.000606\n",
      "zymoseptoria_tritici             0.000454\n",
      "cortinarius_globuliformis        0.000454\n",
      "pyrenophora_tritici-repentis     0.000303\n",
      "puccinia_striiformis-tritici     0.000303\n",
      "clavispora_lusitaniae            0.000303\n",
      "meyerozyma_guillermondii         0.000227\n",
      "kodamaea_ohmeri                  0.000227\n",
      "dothiorella_vidmadera            0.000151\n",
      "asteroma_ccl060                  0.000151\n",
      "cladophialophora_unidentified    0.000151\n",
      "debaryomyces_unidentified        0.000151\n",
      "rhodotorula_mucilaginosa         0.000151\n",
      "oculimacula_yallundae-ccl031     0.000151\n",
      "zygoascus_hellenicus             0.000076\n",
      "candida_metapsilosis             0.000076\n",
      "yamadazyma_scolyti               0.000076\n",
      "yamadazyma_mexicana              0.000076\n",
      "pichia_membranifaciens           0.000076\n",
      "asteroma_ccl068                  0.000076\n",
      "blastobotrys_proliferans         0.000076\n",
      "candida_albicans                 0.000076\n",
      "candida_parapsilosis             0.000076\n",
      "diaporthe_unidentified           0.000076\n",
      "candida_unidentified             0.000076\n",
      "pichia_kudriavzevii              0.000076\n",
      "scedosporium_boydii              0.000076\n",
      "wickerhamomyces_anomalus         0.000076\n",
      "quambalaria_cyanescens           0.000076\n",
      "diaporthe_ccl067                 0.000076\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: aspergillus_flavus\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "aspergillus_flavus               0.990205\n",
      "aspergillus_niger                0.003437\n",
      "aspergillus_unidentified         0.002320\n",
      "penicillium_chrysogenum          0.001718\n",
      "blastobotrys_proliferans         0.000258\n",
      "kodamaea_ohmeri                  0.000258\n",
      "zymoseptoria_tritici             0.000258\n",
      "pyrenophora_tritici-repentis     0.000172\n",
      "rhodotorula_mucilaginosa         0.000172\n",
      "dothiorella_vidmadera            0.000172\n",
      "yamadazyma_scolyti               0.000172\n",
      "entoleuca_unidentified           0.000172\n",
      "quambalaria_cyanescens           0.000086\n",
      "zygoascus_hellenicus             0.000086\n",
      "debaryomyces_unidentified        0.000086\n",
      "cryptococcus_zero                0.000086\n",
      "cladophialophora_unidentified    0.000086\n",
      "candida_metapsilosis             0.000086\n",
      "tuber_brumale                    0.000086\n",
      "yamadazyma_mexicana              0.000086\n",
      "dtype: float64 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ben/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  \"\"\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##########\n",
      "\n",
      "This was the query species: aspergillus_niger\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "aspergillus_niger                0.770105\n",
      "aspergillus_unidentified         0.167564\n",
      "aspergillus_flavus               0.049178\n",
      "penicillium_chrysogenum          0.007175\n",
      "entoleuca_unidentified           0.000747\n",
      "yamadazyma_mexicana              0.000598\n",
      "oculimacula_yallundae-ccl031     0.000598\n",
      "dothiorella_vidmadera            0.000448\n",
      "meyerozyma_guillermondii         0.000448\n",
      "pyrenophora_tritici-repentis     0.000448\n",
      "diaporthe_ccl067                 0.000299\n",
      "asteroma_ccl060                  0.000299\n",
      "pichia_membranifaciens           0.000149\n",
      "yarrowia_lipolytica              0.000149\n",
      "yamadazyma_scolyti               0.000149\n",
      "candida_metapsilosis             0.000149\n",
      "cladophialophora_unidentified    0.000149\n",
      "debaryomyces_unidentified        0.000149\n",
      "tuber_brumale                    0.000149\n",
      "saccharomyces_cerevisiae         0.000149\n",
      "galactomyces_geotrichum          0.000149\n",
      "kodamaea_ohmeri                  0.000149\n",
      "zygoascus_hellenicus             0.000149\n",
      "oculimacula_yallundae-ccl029     0.000149\n",
      "rhodotorula_mucilaginosa         0.000149\n",
      "zymoseptoria_tritici             0.000149\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: pichia_kudriavzevii\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "pichia_kudriavzevii              0.994644\n",
      "pichia_membranifaciens           0.002232\n",
      "clavispora_lusitaniae            0.000446\n",
      "candida_metapsilosis             0.000298\n",
      "candida_orthopsilosis            0.000298\n",
      "tuber_brumale                    0.000223\n",
      "kodamaea_ohmeri                  0.000223\n",
      "cladophialophora_unidentified    0.000223\n",
      "galactomyces_geotrichum          0.000149\n",
      "candida_parapsilosis             0.000149\n",
      "candida_unidentified             0.000149\n",
      "aspergillus_unidentified         0.000074\n",
      "asteroma_ccl060                  0.000074\n",
      "candida_albicans                 0.000074\n",
      "yarrowia_lipolytica              0.000074\n",
      "debaryomyces_unidentified        0.000074\n",
      "yamadazyma_scolyti               0.000074\n",
      "kluyveromyces_unidentified       0.000074\n",
      "meyerozyma_guillermondii         0.000074\n",
      "oculimacula_yallundae-ccl029     0.000074\n",
      "oculimacula_yallundae-ccl031     0.000074\n",
      "saccharomyces_cerevisiae         0.000074\n",
      "yamadazyma_mexicana              0.000074\n",
      "aspergillus_flavus               0.000074\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: pichia_membranifaciens\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "pichia_membranifaciens           0.991393\n",
      "pichia_kudriavzevii              0.002134\n",
      "kodamaea_ohmeri                  0.000782\n",
      "candida_orthopsilosis            0.000569\n",
      "candida_parapsilosis             0.000427\n",
      "wickerhamomyces_anomalus         0.000427\n",
      "candida_metapsilosis             0.000427\n",
      "debaryomyces_unidentified        0.000427\n",
      "galactomyces_geotrichum          0.000427\n",
      "clavispora_lusitaniae            0.000356\n",
      "kluyveromyces_unidentified       0.000285\n",
      "yarrowia_lipolytica              0.000285\n",
      "candida_unidentified             0.000213\n",
      "kluyveromyces_marxianus          0.000213\n",
      "yamadazyma_scolyti               0.000213\n",
      "zygoascus_hellenicus             0.000213\n",
      "meyerozyma_guillermondii         0.000213\n",
      "yamadazyma_mexicana              0.000142\n",
      "quambalaria_cyanescens           0.000142\n",
      "rhodotorula_mucilaginosa         0.000142\n",
      "saccharomyces_cerevisiae         0.000142\n",
      "candida_albicans                 0.000142\n",
      "zymoseptoria_tritici             0.000071\n",
      "cladophialophora_unidentified    0.000071\n",
      "tuber_brumale                    0.000071\n",
      "blastobotrys_proliferans         0.000071\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_albicans\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "candida_albicans                0.416895\n",
      "candida_unidentified            0.278387\n",
      "candida_orthopsilosis           0.124383\n",
      "candida_metapsilosis            0.075699\n",
      "candida_parapsilosis            0.072682\n",
      "meyerozyma_guillermondii        0.006445\n",
      "debaryomyces_unidentified       0.004251\n",
      "galactomyces_geotrichum         0.003017\n",
      "kluyveromyces_marxianus         0.002468\n",
      "yamadazyma_mexicana             0.002057\n",
      "kluyveromyces_unidentified      0.001646\n",
      "kodamaea_ohmeri                 0.001509\n",
      "yamadazyma_scolyti              0.001371\n",
      "pyrenophora_tritici-repentis    0.001097\n",
      "saccharomyces_cerevisiae        0.000960\n",
      "tuber_brumale                   0.000960\n",
      "wickerhamomyces_anomalus        0.000960\n",
      "blastobotrys_proliferans        0.000823\n",
      "clavispora_lusitaniae           0.000823\n",
      "entoleuca_unidentified          0.000549\n",
      "diaporthe_ccl067                0.000549\n",
      "zygoascus_hellenicus            0.000411\n",
      "pichia_membranifaciens          0.000274\n",
      "puccinia_striiformis-tritici    0.000274\n",
      "rhodotorula_mucilaginosa        0.000274\n",
      "oculimacula_yallundae-ccl029    0.000274\n",
      "asteroma_ccl068                 0.000137\n",
      "zymoseptoria_tritici            0.000137\n",
      "cortinarius_globuliformis       0.000137\n",
      "cryptococcus_zero               0.000137\n",
      "pichia_kudriavzevii             0.000137\n",
      "quambalaria_cyanescens          0.000137\n",
      "asteroma_ccl060                 0.000137\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_parapsilosis\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "candida_parapsilosis            0.819925\n",
      "candida_orthopsilosis           0.076185\n",
      "candida_metapsilosis            0.069259\n",
      "candida_albicans                0.008702\n",
      "candida_unidentified            0.008524\n",
      "debaryomyces_unidentified       0.001953\n",
      "meyerozyma_guillermondii        0.001953\n",
      "oculimacula_yallundae-ccl029    0.001243\n",
      "yamadazyma_scolyti              0.001066\n",
      "yamadazyma_mexicana             0.001066\n",
      "galactomyces_geotrichum         0.000888\n",
      "kluyveromyces_marxianus         0.000888\n",
      "wickerhamomyces_anomalus        0.000888\n",
      "kodamaea_ohmeri                 0.000888\n",
      "entoleuca_unidentified          0.000710\n",
      "kluyveromyces_unidentified      0.000710\n",
      "aspergillus_niger               0.000533\n",
      "scedosporium_boydii             0.000533\n",
      "saccharomyces_cerevisiae        0.000533\n",
      "zygoascus_hellenicus            0.000355\n",
      "tuber_brumale                   0.000355\n",
      "pichia_kudriavzevii             0.000355\n",
      "fusarium_oxysporum              0.000355\n",
      "pichia_membranifaciens          0.000355\n",
      "pyrenophora_tritici-repentis    0.000178\n",
      "asteroma_ccl060                 0.000178\n",
      "blastobotrys_proliferans        0.000178\n",
      "yarrowia_lipolytica             0.000178\n",
      "cortinarius_globuliformis       0.000178\n",
      "cryptococcus_zero               0.000178\n",
      "diaporthe_unidentified          0.000178\n",
      "dothiorella_vidmadera           0.000178\n",
      "oculimacula_yallundae-ccl031    0.000178\n",
      "clavispora_lusitaniae           0.000178\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_orthopsilosis\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "candida_orthopsilosis            0.595565\n",
      "candida_metapsilosis             0.217829\n",
      "candida_parapsilosis             0.131310\n",
      "candida_unidentified             0.018238\n",
      "meyerozyma_guillermondii         0.011380\n",
      "candida_albicans                 0.010213\n",
      "debaryomyces_unidentified        0.003356\n",
      "kodamaea_ohmeri                  0.001459\n",
      "yamadazyma_scolyti               0.001167\n",
      "yamadazyma_mexicana              0.001167\n",
      "galactomyces_geotrichum          0.001021\n",
      "wickerhamomyces_anomalus         0.000875\n",
      "oculimacula_yallundae-ccl029     0.000730\n",
      "blastobotrys_proliferans         0.000730\n",
      "saccharomyces_cerevisiae         0.000584\n",
      "clavispora_lusitaniae            0.000438\n",
      "yarrowia_lipolytica              0.000438\n",
      "kluyveromyces_marxianus          0.000438\n",
      "entoleuca_unidentified           0.000292\n",
      "pichia_kudriavzevii              0.000292\n",
      "kluyveromyces_unidentified       0.000292\n",
      "tuber_brumale                    0.000292\n",
      "cladophialophora_unidentified    0.000292\n",
      "asteroma_ccl060                  0.000146\n",
      "zymoseptoria_tritici             0.000146\n",
      "cryptococcus_zero                0.000146\n",
      "diaporthe_ccl067                 0.000146\n",
      "dothiorella_vidmadera            0.000146\n",
      "zygoascus_hellenicus             0.000146\n",
      "pichia_membranifaciens           0.000146\n",
      "pyrenophora_tritici-repentis     0.000146\n",
      "quambalaria_cyanescens           0.000146\n",
      "scedosporium_boydii              0.000146\n",
      "aspergillus_niger                0.000146\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_metapsilosis\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "candida_metapsilosis             0.425555\n",
      "candida_orthopsilosis            0.349892\n",
      "candida_parapsilosis             0.166892\n",
      "candida_unidentified             0.020303\n",
      "candida_albicans                 0.012723\n",
      "kodamaea_ohmeri                  0.005143\n",
      "meyerozyma_guillermondii         0.003249\n",
      "debaryomyces_unidentified        0.003113\n",
      "diaporthe_unidentified           0.002166\n",
      "fusarium_oxysporum               0.001760\n",
      "yamadazyma_scolyti               0.000947\n",
      "galactomyces_geotrichum          0.000947\n",
      "oculimacula_yallundae-ccl029     0.000812\n",
      "yamadazyma_mexicana              0.000812\n",
      "kluyveromyces_unidentified       0.000677\n",
      "tuber_brumale                    0.000677\n",
      "pichia_kudriavzevii              0.000541\n",
      "blastobotrys_proliferans         0.000541\n",
      "clavispora_lusitaniae            0.000541\n",
      "wickerhamomyces_anomalus         0.000406\n",
      "penicillium_chrysogenum          0.000271\n",
      "saccharomyces_cerevisiae         0.000271\n",
      "entoleuca_unidentified           0.000271\n",
      "kluyveromyces_marxianus          0.000271\n",
      "asteroma_ccl060                  0.000135\n",
      "zygoascus_hellenicus             0.000135\n",
      "cladophialophora_unidentified    0.000135\n",
      "cortinarius_globuliformis        0.000135\n",
      "yarrowia_lipolytica              0.000135\n",
      "puccinia_striiformis-tritici     0.000135\n",
      "quambalaria_cyanescens           0.000135\n",
      "rhodotorula_mucilaginosa         0.000135\n",
      "aspergillus_flavus               0.000135\n",
      "dtype: float64 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "###this is running the reads against the full database\n",
    "for species, hit_fn in new_db_mapping_fn.items():\n",
    "    mapping_results(hit_fn, species)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##########\n",
      "\n",
      "This was the query species: penicillium_chrysogenum\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "penicillium_chrysogenum          0.989973\n",
      "aspergillus_unidentified         0.002302\n",
      "aspergillus_flavus               0.001783\n",
      "kluyveromyces_marxianus          0.000743\n",
      "tuber_brumale                    0.000594\n",
      "zymoseptoria_tritici             0.000520\n",
      "kluyveromyces_unidentified       0.000446\n",
      "cortinarius_globuliformis        0.000446\n",
      "puccinia_striiformis-tritici     0.000371\n",
      "pyrenophora_tritici-repentis     0.000371\n",
      "rhodotorula_mucilaginosa         0.000297\n",
      "meyerozyma_guillermondii         0.000223\n",
      "kodamaea_ohmeri                  0.000223\n",
      "clavispora_lusitaniae            0.000223\n",
      "cladophialophora_unidentified    0.000223\n",
      "asteroma_ccl068                  0.000149\n",
      "debaryomyces_unidentified        0.000149\n",
      "oculimacula_yallundae-ccl031     0.000149\n",
      "quambalaria_cyanescens           0.000074\n",
      "candida_parapsilosis             0.000074\n",
      "scedosporium_boydii              0.000074\n",
      "asteroma_ccl060                  0.000074\n",
      "blastobotrys_proliferans         0.000074\n",
      "pichia_membranifaciens           0.000074\n",
      "candida_unidentified             0.000074\n",
      "pichia_kudriavzevii              0.000074\n",
      "diaporthe_unidentified           0.000074\n",
      "wickerhamomyces_anomalus         0.000074\n",
      "diaporthe_ccl067                 0.000074\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: aspergillus_flavus\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "aspergillus_flavus              0.995077\n",
      "aspergillus_unidentified        0.002121\n",
      "penicillium_chrysogenum         0.001136\n",
      "blastobotrys_proliferans        0.000227\n",
      "pyrenophora_tritici-repentis    0.000151\n",
      "kodamaea_ohmeri                 0.000151\n",
      "entoleuca_unidentified          0.000151\n",
      "dothiorella_vidmadera           0.000151\n",
      "cryptococcus_zero               0.000151\n",
      "zymoseptoria_tritici            0.000151\n",
      "yamadazyma_mexicana             0.000076\n",
      "tuber_brumale                   0.000076\n",
      "rhodotorula_mucilaginosa        0.000076\n",
      "quambalaria_cyanescens          0.000076\n",
      "zygoascus_hellenicus            0.000076\n",
      "debaryomyces_unidentified       0.000076\n",
      "yamadazyma_scolyti              0.000076\n",
      "dtype: float64 \n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ben/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  \"\"\"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "##########\n",
      "\n",
      "This was the query species: aspergillus_niger\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "aspergillus_unidentified         0.940982\n",
      "aspergillus_flavus               0.040535\n",
      "penicillium_chrysogenum          0.011217\n",
      "dothiorella_vidmadera            0.001530\n",
      "zymoseptoria_tritici             0.001147\n",
      "meyerozyma_guillermondii         0.000510\n",
      "pyrenophora_tritici-repentis     0.000510\n",
      "entoleuca_unidentified           0.000510\n",
      "cladophialophora_unidentified    0.000382\n",
      "tuber_brumale                    0.000255\n",
      "diaporthe_ccl067                 0.000255\n",
      "yamadazyma_mexicana              0.000255\n",
      "zygoascus_hellenicus             0.000255\n",
      "oculimacula_yallundae-ccl031     0.000255\n",
      "puccinia_striiformis-tritici     0.000255\n",
      "oculimacula_yallundae-ccl029     0.000255\n",
      "saccharomyces_cerevisiae         0.000127\n",
      "diaporthe_unidentified           0.000127\n",
      "wickerhamomyces_anomalus         0.000127\n",
      "debaryomyces_unidentified        0.000127\n",
      "candida_albicans                 0.000127\n",
      "asteroma_ccl068                  0.000127\n",
      "yamadazyma_scolyti               0.000127\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: pichia_kudriavzevii\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "pichia_kudriavzevii              0.995168\n",
      "pichia_membranifaciens           0.002081\n",
      "clavispora_lusitaniae            0.000446\n",
      "tuber_brumale                    0.000297\n",
      "candida_parapsilosis             0.000297\n",
      "zymoseptoria_tritici             0.000149\n",
      "blastobotrys_proliferans         0.000149\n",
      "cladophialophora_unidentified    0.000149\n",
      "galactomyces_geotrichum          0.000149\n",
      "debaryomyces_unidentified        0.000149\n",
      "aspergillus_unidentified         0.000074\n",
      "asteroma_ccl060                  0.000074\n",
      "candida_albicans                 0.000074\n",
      "candida_unidentified             0.000074\n",
      "kodamaea_ohmeri                  0.000074\n",
      "zygoascus_hellenicus             0.000074\n",
      "meyerozyma_guillermondii         0.000074\n",
      "oculimacula_yallundae-ccl029     0.000074\n",
      "saccharomyces_cerevisiae         0.000074\n",
      "wickerhamomyces_anomalus         0.000074\n",
      "yamadazyma_scolyti               0.000074\n",
      "yarrowia_lipolytica              0.000074\n",
      "aspergillus_flavus               0.000074\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: pichia_membranifaciens\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "pichia_membranifaciens           0.991194\n",
      "pichia_kudriavzevii              0.002131\n",
      "kodamaea_ohmeri                  0.000781\n",
      "candida_parapsilosis             0.000639\n",
      "debaryomyces_unidentified        0.000568\n",
      "galactomyces_geotrichum          0.000497\n",
      "meyerozyma_guillermondii         0.000497\n",
      "clavispora_lusitaniae            0.000426\n",
      "candida_albicans                 0.000426\n",
      "wickerhamomyces_anomalus         0.000426\n",
      "candida_unidentified             0.000355\n",
      "kluyveromyces_unidentified       0.000355\n",
      "zygoascus_hellenicus             0.000213\n",
      "yamadazyma_mexicana              0.000213\n",
      "kluyveromyces_marxianus          0.000213\n",
      "yarrowia_lipolytica              0.000213\n",
      "rhodotorula_mucilaginosa         0.000142\n",
      "quambalaria_cyanescens           0.000142\n",
      "yamadazyma_scolyti               0.000142\n",
      "zymoseptoria_tritici             0.000071\n",
      "oculimacula_yallundae-ccl029     0.000071\n",
      "cladophialophora_unidentified    0.000071\n",
      "saccharomyces_cerevisiae         0.000071\n",
      "tuber_brumale                    0.000071\n",
      "blastobotrys_proliferans         0.000071\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_albicans\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "candida_albicans                0.484864\n",
      "candida_unidentified            0.315478\n",
      "candida_parapsilosis            0.160305\n",
      "meyerozyma_guillermondii        0.009608\n",
      "debaryomyces_unidentified       0.005791\n",
      "yamadazyma_mexicana             0.002632\n",
      "kluyveromyces_unidentified      0.002369\n",
      "kluyveromyces_marxianus         0.002369\n",
      "yamadazyma_scolyti              0.002369\n",
      "galactomyces_geotrichum         0.001974\n",
      "kodamaea_ohmeri                 0.001711\n",
      "saccharomyces_cerevisiae        0.001579\n",
      "wickerhamomyces_anomalus        0.001448\n",
      "blastobotrys_proliferans        0.001316\n",
      "clavispora_lusitaniae           0.000921\n",
      "pyrenophora_tritici-repentis    0.000790\n",
      "entoleuca_unidentified          0.000658\n",
      "pichia_membranifaciens          0.000526\n",
      "tuber_brumale                   0.000526\n",
      "oculimacula_yallundae-ccl029    0.000395\n",
      "asteroma_ccl060                 0.000263\n",
      "diaporthe_ccl067                0.000263\n",
      "cryptococcus_zero               0.000263\n",
      "zygoascus_hellenicus            0.000263\n",
      "asteroma_ccl068                 0.000132\n",
      "zymoseptoria_tritici            0.000132\n",
      "cortinarius_globuliformis       0.000132\n",
      "diaporthe_unidentified          0.000132\n",
      "dothiorella_vidmadera           0.000132\n",
      "pichia_kudriavzevii             0.000132\n",
      "quambalaria_cyanescens          0.000132\n",
      "rhodotorula_mucilaginosa        0.000132\n",
      "scedosporium_boydii             0.000132\n",
      "aspergillus_flavus              0.000132\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_parapsilosis\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "candida_parapsilosis             0.985721\n",
      "candida_unidentified             0.002487\n",
      "candida_albicans                 0.001925\n",
      "debaryomyces_unidentified        0.000963\n",
      "meyerozyma_guillermondii         0.000882\n",
      "kodamaea_ohmeri                  0.000722\n",
      "clavispora_lusitaniae            0.000562\n",
      "blastobotrys_proliferans         0.000562\n",
      "yamadazyma_mexicana              0.000562\n",
      "galactomyces_geotrichum          0.000481\n",
      "kluyveromyces_unidentified       0.000481\n",
      "kluyveromyces_marxianus          0.000481\n",
      "yamadazyma_scolyti               0.000481\n",
      "pichia_membranifaciens           0.000481\n",
      "saccharomyces_cerevisiae         0.000401\n",
      "tuber_brumale                    0.000401\n",
      "pichia_kudriavzevii              0.000321\n",
      "wickerhamomyces_anomalus         0.000321\n",
      "cladophialophora_unidentified    0.000241\n",
      "zymoseptoria_tritici             0.000241\n",
      "dothiorella_vidmadera            0.000241\n",
      "oculimacula_yallundae-ccl029     0.000241\n",
      "yarrowia_lipolytica              0.000160\n",
      "diaporthe_unidentified           0.000080\n",
      "cryptococcus_zero                0.000080\n",
      "entoleuca_unidentified           0.000080\n",
      "zygoascus_hellenicus             0.000080\n",
      "pyrenophora_tritici-repentis     0.000080\n",
      "asteroma_ccl060                  0.000080\n",
      "aspergillus_unidentified         0.000080\n",
      "aspergillus_flavus               0.000080\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_orthopsilosis\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "candida_parapsilosis             0.939677\n",
      "meyerozyma_guillermondii         0.017758\n",
      "candida_unidentified             0.011929\n",
      "candida_albicans                 0.007998\n",
      "debaryomyces_unidentified        0.004609\n",
      "yamadazyma_scolyti               0.002982\n",
      "yamadazyma_mexicana              0.002847\n",
      "wickerhamomyces_anomalus         0.001898\n",
      "kodamaea_ohmeri                  0.001220\n",
      "clavispora_lusitaniae            0.001084\n",
      "blastobotrys_proliferans         0.001084\n",
      "tuber_brumale                    0.001084\n",
      "galactomyces_geotrichum          0.000813\n",
      "saccharomyces_cerevisiae         0.000678\n",
      "kluyveromyces_marxianus          0.000678\n",
      "kluyveromyces_unidentified       0.000678\n",
      "zygoascus_hellenicus             0.000407\n",
      "dothiorella_vidmadera            0.000407\n",
      "pichia_membranifaciens           0.000271\n",
      "entoleuca_unidentified           0.000271\n",
      "cladophialophora_unidentified    0.000271\n",
      "asteroma_ccl068                  0.000136\n",
      "zymoseptoria_tritici             0.000136\n",
      "diaporthe_ccl067                 0.000136\n",
      "diaporthe_unidentified           0.000136\n",
      "penicillium_chrysogenum          0.000136\n",
      "pichia_kudriavzevii              0.000136\n",
      "quambalaria_cyanescens           0.000136\n",
      "rhodotorula_mucilaginosa         0.000136\n",
      "yarrowia_lipolytica              0.000136\n",
      "asteroma_ccl060                  0.000136\n",
      "dtype: float64 \n",
      "\n",
      "##########\n",
      "\n",
      "This was the query species: candida_metapsilosis\n",
      "\n",
      "These are the results:\n",
      "tname\n",
      "candida_parapsilosis             0.965001\n",
      "candida_unidentified             0.007355\n",
      "candida_albicans                 0.005960\n",
      "meyerozyma_guillermondii         0.004692\n",
      "debaryomyces_unidentified        0.003424\n",
      "kodamaea_ohmeri                  0.002156\n",
      "yamadazyma_scolyti               0.001395\n",
      "yamadazyma_mexicana              0.001268\n",
      "wickerhamomyces_anomalus         0.001268\n",
      "kluyveromyces_marxianus          0.000888\n",
      "blastobotrys_proliferans         0.000888\n",
      "kluyveromyces_unidentified       0.000888\n",
      "saccharomyces_cerevisiae         0.000761\n",
      "clavispora_lusitaniae            0.000634\n",
      "galactomyces_geotrichum          0.000634\n",
      "tuber_brumale                    0.000507\n",
      "zymoseptoria_tritici             0.000380\n",
      "oculimacula_yallundae-ccl031     0.000380\n",
      "pichia_kudriavzevii              0.000254\n",
      "pyrenophora_tritici-repentis     0.000254\n",
      "yarrowia_lipolytica              0.000127\n",
      "fusarium_oxysporum               0.000127\n",
      "entoleuca_unidentified           0.000127\n",
      "cladophialophora_unidentified    0.000127\n",
      "pichia_membranifaciens           0.000127\n",
      "quambalaria_cyanescens           0.000127\n",
      "rhodotorula_mucilaginosa         0.000127\n",
      "asteroma_ccl068                  0.000127\n",
      "dtype: float64 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "###this is running against a database that have ['Candida_orthopsilosis', 'Candida_metapsilosis', 'Aspergillus_niger'] deleted\n",
    "for species, hit_fn in sub_db_mapping_fn.items():\n",
    "    mapping_results(hit_fn, species)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
